// @fattureincloud/fattureincloud-js-sdk/dist/model/CashbookEntry.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _CashbookEntryDocument = _interopRequireDefault(require("./CashbookEntryDocument"));
var _CashbookEntryKind = _interopRequireDefault(require("./CashbookEntryKind"));
var _CashbookEntryType = _interopRequireDefault(require("./CashbookEntryType"));
var _PaymentAccount = _interopRequireDefault(require("./PaymentAccount"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The CashbookEntry model module.
 * @module model/CashbookEntry
 * @version 2.0.25
 */
var CashbookEntry = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>CashbookEntry</code>.
     * @alias module:model/CashbookEntry
     */
  function CashbookEntry() {
    _classCallCheck(this, CashbookEntry);
    CashbookEntry.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(CashbookEntry, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>CashbookEntry</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/CashbookEntry} obj Optional instance to populate.
       * @return {module:model/CashbookEntry} The populated <code>CashbookEntry</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CashbookEntry();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'String');
        }
        if (data.hasOwnProperty('date')) {
          obj.date = _ApiClient["default"].convertToType(data.date, 'Date');
        }
        if (data.hasOwnProperty('description')) {
          obj.description = _ApiClient["default"].convertToType(data.description, 'String');
        }
        if (data.hasOwnProperty('kind')) {
          obj.kind = _CashbookEntryKind["default"].constructFromObject(data.kind);
        }
        if (data.hasOwnProperty('type')) {
          obj.type = _CashbookEntryType["default"].constructFromObject(data.type);
        }
        if (data.hasOwnProperty('entity_name')) {
          obj.entity_name = _ApiClient["default"].convertToType(data.entity_name, 'String');
        }
        if (data.hasOwnProperty('document')) {
          obj.document = _CashbookEntryDocument["default"].constructFromObject(data.document);
        }
        if (data.hasOwnProperty('amount_in')) {
          obj.amount_in = _ApiClient["default"].convertToType(data.amount_in, 'Number');
        }
        if (data.hasOwnProperty('payment_account_in')) {
          obj.payment_account_in = _PaymentAccount["default"].constructFromObject(data.payment_account_in);
        }
        if (data.hasOwnProperty('amount_out')) {
          obj.amount_out = _ApiClient["default"].convertToType(data.amount_out, 'Number');
        }
        if (data.hasOwnProperty('payment_account_out')) {
          obj.payment_account_out = _PaymentAccount["default"].constructFromObject(data.payment_account_out);
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>CashbookEntry</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CashbookEntry</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.id && !(typeof data.id === 'string' || data.id instanceof String)) {
        throw new Error('Expected the field `id` to be a primitive type in the JSON string but got ' + data.id);
      }
      // ensure the json data is a string
      if (data.description && !(typeof data.description === 'string' || data.description instanceof String)) {
        throw new Error('Expected the field `description` to be a primitive type in the JSON string but got ' + data.description);
      }
      // ensure the json data is a string
      if (data.entity_name && !(typeof data.entity_name === 'string' || data.entity_name instanceof String)) {
        throw new Error('Expected the field `entity_name` to be a primitive type in the JSON string but got ' + data.entity_name);
      }
      // validate the optional field `document`
      if (data.document) {
        // data not null
        _CashbookEntryDocument["default"].validateJSON(data.document);
      }
      // validate the optional field `payment_account_in`
      if (data.payment_account_in) {
        // data not null
        _PaymentAccount["default"].validateJSON(data.payment_account_in);
      }
      // validate the optional field `payment_account_out`
      if (data.payment_account_out) {
        // data not null
        _PaymentAccount["default"].validateJSON(data.payment_account_out);
      }
      return true;
    }
  }]);
  return CashbookEntry;
}();
/**
 * Cashbook id
 * @member {String} id
 */
CashbookEntry.prototype.id = undefined;

/**
 * Cashbook date
 * @member {Date} date
 */
CashbookEntry.prototype.date = undefined;

/**
 * Cashbook description
 * @member {String} description
 */
CashbookEntry.prototype.description = undefined;

/**
 * @member {module:model/CashbookEntryKind} kind
 */
CashbookEntry.prototype.kind = undefined;

/**
 * @member {module:model/CashbookEntryType} type
 */
CashbookEntry.prototype.type = undefined;

/**
 * Cashbook entity name
 * @member {String} entity_name
 */
CashbookEntry.prototype.entity_name = undefined;

/**
 * @member {module:model/CashbookEntryDocument} document
 */
CashbookEntry.prototype.document = undefined;

/**
 * [Only for cashbook entry in] Cashbook total amount in
 * @member {Number} amount_in
 */
CashbookEntry.prototype.amount_in = undefined;

/**
 * @member {module:model/PaymentAccount} payment_account_in
 */
CashbookEntry.prototype.payment_account_in = undefined;

/**
 * [Only for cashbook entry out] Cashbook total amount out
 * @member {Number} amount_out
 */
CashbookEntry.prototype.amount_out = undefined;

/**
 * @member {module:model/PaymentAccount} payment_account_out
 */
CashbookEntry.prototype.payment_account_out = undefined;
var _default = exports["default"] = CashbookEntry;
