// @fattureincloud/fattureincloud-js-sdk/dist/model/CompanyInfoPlanInfoFunctions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The CompanyInfoPlanInfoFunctions model module.
 * @module model/CompanyInfoPlanInfoFunctions
 * @version 2.0.25
 */
var CompanyInfoPlanInfoFunctions = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>CompanyInfoPlanInfoFunctions</code>.
     * Access to functions for this company.
     * @alias module:model/CompanyInfoPlanInfoFunctions
     */
  function CompanyInfoPlanInfoFunctions() {
    _classCallCheck(this, CompanyInfoPlanInfoFunctions);
    CompanyInfoPlanInfoFunctions.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(CompanyInfoPlanInfoFunctions, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>CompanyInfoPlanInfoFunctions</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/CompanyInfoPlanInfoFunctions} obj Optional instance to populate.
       * @return {module:model/CompanyInfoPlanInfoFunctions} The populated <code>CompanyInfoPlanInfoFunctions</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CompanyInfoPlanInfoFunctions();
        if (data.hasOwnProperty('archive')) {
          obj.archive = _ApiClient["default"].convertToType(data.archive, 'Boolean');
        }
        if (data.hasOwnProperty('cerved')) {
          obj.cerved = _ApiClient["default"].convertToType(data.cerved, 'Boolean');
        }
        if (data.hasOwnProperty('document_attachments')) {
          obj.document_attachments = _ApiClient["default"].convertToType(data.document_attachments, 'Boolean');
        }
        if (data.hasOwnProperty('e_invoice')) {
          obj.e_invoice = _ApiClient["default"].convertToType(data.e_invoice, 'Boolean');
        }
        if (data.hasOwnProperty('genius')) {
          obj.genius = _ApiClient["default"].convertToType(data.genius, 'Boolean');
        }
        if (data.hasOwnProperty('mail_tracking')) {
          obj.mail_tracking = _ApiClient["default"].convertToType(data.mail_tracking, 'Boolean');
        }
        if (data.hasOwnProperty('payment_notifications')) {
          obj.payment_notifications = _ApiClient["default"].convertToType(data.payment_notifications, 'Boolean');
        }
        if (data.hasOwnProperty('paypal')) {
          obj.paypal = _ApiClient["default"].convertToType(data.paypal, 'Boolean');
        }
        if (data.hasOwnProperty('receipts')) {
          obj.receipts = _ApiClient["default"].convertToType(data.receipts, 'Boolean');
        }
        if (data.hasOwnProperty('recurring')) {
          obj.recurring = _ApiClient["default"].convertToType(data.recurring, 'Boolean');
        }
        if (data.hasOwnProperty('smtp')) {
          obj.smtp = _ApiClient["default"].convertToType(data.smtp, 'Boolean');
        }
        if (data.hasOwnProperty('sofort')) {
          obj.sofort = _ApiClient["default"].convertToType(data.sofort, 'Boolean');
        }
        if (data.hasOwnProperty('stock')) {
          obj.stock = _ApiClient["default"].convertToType(data.stock, 'Boolean');
        }
        if (data.hasOwnProperty('subaccounts')) {
          obj.subaccounts = _ApiClient["default"].convertToType(data.subaccounts, 'Boolean');
        }
        if (data.hasOwnProperty('tessera_sanitaria')) {
          obj.tessera_sanitaria = _ApiClient["default"].convertToType(data.tessera_sanitaria, 'Boolean');
        }
        if (data.hasOwnProperty('ts_digital')) {
          obj.ts_digital = _ApiClient["default"].convertToType(data.ts_digital, 'Boolean');
        }
        if (data.hasOwnProperty('ts_invoice_trading')) {
          obj.ts_invoice_trading = _ApiClient["default"].convertToType(data.ts_invoice_trading, 'Boolean');
        }
        if (data.hasOwnProperty('ts_pay')) {
          obj.ts_pay = _ApiClient["default"].convertToType(data.ts_pay, 'Boolean');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>CompanyInfoPlanInfoFunctions</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CompanyInfoPlanInfoFunctions</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      return true;
    }
  }]);
  return CompanyInfoPlanInfoFunctions;
}();
/**
 * @member {Boolean} archive
 */
CompanyInfoPlanInfoFunctions.prototype.archive = undefined;

/**
 * @member {Boolean} cerved
 */
CompanyInfoPlanInfoFunctions.prototype.cerved = undefined;

/**
 * @member {Boolean} document_attachments
 */
CompanyInfoPlanInfoFunctions.prototype.document_attachments = undefined;

/**
 * @member {Boolean} e_invoice
 */
CompanyInfoPlanInfoFunctions.prototype.e_invoice = undefined;

/**
 * @member {Boolean} genius
 */
CompanyInfoPlanInfoFunctions.prototype.genius = undefined;

/**
 * @member {Boolean} mail_tracking
 */
CompanyInfoPlanInfoFunctions.prototype.mail_tracking = undefined;

/**
 * @member {Boolean} payment_notifications
 */
CompanyInfoPlanInfoFunctions.prototype.payment_notifications = undefined;

/**
 * @member {Boolean} paypal
 */
CompanyInfoPlanInfoFunctions.prototype.paypal = undefined;

/**
 * @member {Boolean} receipts
 */
CompanyInfoPlanInfoFunctions.prototype.receipts = undefined;

/**
 * @member {Boolean} recurring
 */
CompanyInfoPlanInfoFunctions.prototype.recurring = undefined;

/**
 * @member {Boolean} smtp
 */
CompanyInfoPlanInfoFunctions.prototype.smtp = undefined;

/**
 * @member {Boolean} sofort
 */
CompanyInfoPlanInfoFunctions.prototype.sofort = undefined;

/**
 * @member {Boolean} stock
 */
CompanyInfoPlanInfoFunctions.prototype.stock = undefined;

/**
 * @member {Boolean} subaccounts
 */
CompanyInfoPlanInfoFunctions.prototype.subaccounts = undefined;

/**
 * @member {Boolean} tessera_sanitaria
 */
CompanyInfoPlanInfoFunctions.prototype.tessera_sanitaria = undefined;

/**
 * @member {Boolean} ts_digital
 */
CompanyInfoPlanInfoFunctions.prototype.ts_digital = undefined;

/**
 * @member {Boolean} ts_invoice_trading
 */
CompanyInfoPlanInfoFunctions.prototype.ts_invoice_trading = undefined;

/**
 * @member {Boolean} ts_pay
 */
CompanyInfoPlanInfoFunctions.prototype.ts_pay = undefined;
var _default = exports["default"] = CompanyInfoPlanInfoFunctions;
