// @fattureincloud/fattureincloud-js-sdk/dist/model/CompanyInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _CompanyInfoAccessInfo = _interopRequireDefault(require("./CompanyInfoAccessInfo"));
var _CompanyInfoPlanInfo = _interopRequireDefault(require("./CompanyInfoPlanInfo"));
var _CompanyType = _interopRequireDefault(require("./CompanyType"));
var _FattureInCloudPlanType = _interopRequireDefault(require("./FattureInCloudPlanType"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The CompanyInfo model module.
 * @module model/CompanyInfo
 * @version 2.0.25
 */
var CompanyInfo = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>CompanyInfo</code>.
     * @alias module:model/CompanyInfo
     */
  function CompanyInfo() {
    _classCallCheck(this, CompanyInfo);
    CompanyInfo.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(CompanyInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>CompanyInfo</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/CompanyInfo} obj Optional instance to populate.
       * @return {module:model/CompanyInfo} The populated <code>CompanyInfo</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CompanyInfo();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'Number');
        }
        if (data.hasOwnProperty('name')) {
          obj.name = _ApiClient["default"].convertToType(data.name, 'String');
        }
        if (data.hasOwnProperty('email')) {
          obj.email = _ApiClient["default"].convertToType(data.email, 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj.type = _CompanyType["default"].constructFromObject(data.type);
        }
        if (data.hasOwnProperty('access_info')) {
          obj.access_info = _CompanyInfoAccessInfo["default"].constructFromObject(data.access_info);
        }
        if (data.hasOwnProperty('fic_license_expire')) {
          obj.fic_license_expire = _ApiClient["default"].convertToType(data.fic_license_expire, 'Date');
        }
        if (data.hasOwnProperty('fic_plan_name')) {
          obj.fic_plan_name = _FattureInCloudPlanType["default"].constructFromObject(data.fic_plan_name);
        }
        if (data.hasOwnProperty('plan_info')) {
          obj.plan_info = _CompanyInfoPlanInfo["default"].constructFromObject(data.plan_info);
        }
        if (data.hasOwnProperty('accountant_id')) {
          obj.accountant_id = _ApiClient["default"].convertToType(data.accountant_id, 'Number');
        }
        if (data.hasOwnProperty('is_accountant')) {
          obj.is_accountant = _ApiClient["default"].convertToType(data.is_accountant, 'Boolean');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>CompanyInfo</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CompanyInfo</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.name && !(typeof data.name === 'string' || data.name instanceof String)) {
        throw new Error('Expected the field `name` to be a primitive type in the JSON string but got ' + data.name);
      }
      // ensure the json data is a string
      if (data.email && !(typeof data.email === 'string' || data.email instanceof String)) {
        throw new Error('Expected the field `email` to be a primitive type in the JSON string but got ' + data.email);
      }
      // validate the optional field `access_info`
      if (data.access_info) {
        // data not null
        _CompanyInfoAccessInfo["default"].validateJSON(data.access_info);
      }
      // validate the optional field `plan_info`
      if (data.plan_info) {
        // data not null
        _CompanyInfoPlanInfo["default"].validateJSON(data.plan_info);
      }
      return true;
    }
  }]);
  return CompanyInfo;
}();
/**
 * Company id
 * @member {Number} id
 */
CompanyInfo.prototype.id = undefined;

/**
 * Company name
 * @member {String} name
 */
CompanyInfo.prototype.name = undefined;

/**
 * Company email
 * @member {String} email
 */
CompanyInfo.prototype.email = undefined;

/**
 * @member {module:model/CompanyType} type
 */
CompanyInfo.prototype.type = undefined;

/**
 * @member {module:model/CompanyInfoAccessInfo} access_info
 */
CompanyInfo.prototype.access_info = undefined;

/**
 * @member {Date} fic_license_expire
 */
CompanyInfo.prototype.fic_license_expire = undefined;

/**
 * @member {module:model/FattureInCloudPlanType} fic_plan_name
 */
CompanyInfo.prototype.fic_plan_name = undefined;

/**
 * @member {module:model/CompanyInfoPlanInfo} plan_info
 */
CompanyInfo.prototype.plan_info = undefined;

/**
 * Company accountant id
 * @member {Number} accountant_id
 */
CompanyInfo.prototype.accountant_id = undefined;

/**
 * Is the logged account an accountant.
 * @member {Boolean} is_accountant
 */
CompanyInfo.prototype.is_accountant = undefined;
var _default = exports["default"] = CompanyInfo;
