// @fattureincloud/fattureincloud-js-sdk/dist/model/Currency.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The Currency model module.
 * @module model/Currency
 * @version 2.0.25
 */
var Currency = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>Currency</code>.
     * @alias module:model/Currency
     */
  function Currency() {
    _classCallCheck(this, Currency);
    Currency.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(Currency, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>Currency</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/Currency} obj Optional instance to populate.
       * @return {module:model/Currency} The populated <code>Currency</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Currency();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'String');
        }
        if (data.hasOwnProperty('symbol')) {
          obj.symbol = _ApiClient["default"].convertToType(data.symbol, 'String');
        }
        if (data.hasOwnProperty('exchange_rate')) {
          obj.exchange_rate = _ApiClient["default"].convertToType(data.exchange_rate, 'String');
        }
        if (data.hasOwnProperty('html_symbol')) {
          obj.html_symbol = _ApiClient["default"].convertToType(data.html_symbol, 'String');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>Currency</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Currency</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.id && !(typeof data.id === 'string' || data.id instanceof String)) {
        throw new Error('Expected the field `id` to be a primitive type in the JSON string but got ' + data.id);
      }
      // ensure the json data is a string
      if (data.symbol && !(typeof data.symbol === 'string' || data.symbol instanceof String)) {
        throw new Error('Expected the field `symbol` to be a primitive type in the JSON string but got ' + data.symbol);
      }
      // ensure the json data is a string
      if (data.exchange_rate && !(typeof data.exchange_rate === 'string' || data.exchange_rate instanceof String)) {
        throw new Error('Expected the field `exchange_rate` to be a primitive type in the JSON string but got ' + data.exchange_rate);
      }
      // ensure the json data is a string
      if (data.html_symbol && !(typeof data.html_symbol === 'string' || data.html_symbol instanceof String)) {
        throw new Error('Expected the field `html_symbol` to be a primitive type in the JSON string but got ' + data.html_symbol);
      }
      return true;
    }
  }]);
  return Currency;
}();
/**
 * Currency code
 * @member {String} id
 */
Currency.prototype.id = undefined;

/**
 * Currency symbol
 * @member {String} symbol
 */
Currency.prototype.symbol = undefined;

/**
 * Currency exchange rate (EUR to this)
 * @member {String} exchange_rate
 */
Currency.prototype.exchange_rate = undefined;

/**
 * Currency html code
 * @member {String} html_symbol
 */
Currency.prototype.html_symbol = undefined;
var _default = exports["default"] = Currency;
