// @fattureincloud/fattureincloud-js-sdk/dist/model/DetailedCountry.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The DetailedCountry model module.
 * @module model/DetailedCountry
 * @version 2.0.25
 */
var DetailedCountry = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>DetailedCountry</code>.
     * @alias module:model/DetailedCountry
     */
  function DetailedCountry() {
    _classCallCheck(this, DetailedCountry);
    DetailedCountry.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(DetailedCountry, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>DetailedCountry</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/DetailedCountry} obj Optional instance to populate.
       * @return {module:model/DetailedCountry} The populated <code>DetailedCountry</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DetailedCountry();
        if (data.hasOwnProperty('name')) {
          obj.name = _ApiClient["default"].convertToType(data.name, 'String');
        }
        if (data.hasOwnProperty('settings_name')) {
          obj.settings_name = _ApiClient["default"].convertToType(data.settings_name, 'String');
        }
        if (data.hasOwnProperty('iso')) {
          obj.iso = _ApiClient["default"].convertToType(data.iso, 'String');
        }
        if (data.hasOwnProperty('fiscal_iso')) {
          obj.fiscal_iso = _ApiClient["default"].convertToType(data.fiscal_iso, 'String');
        }
        if (data.hasOwnProperty('uic')) {
          obj.uic = _ApiClient["default"].convertToType(data.uic, 'String');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>DetailedCountry</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>DetailedCountry</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.name && !(typeof data.name === 'string' || data.name instanceof String)) {
        throw new Error('Expected the field `name` to be a primitive type in the JSON string but got ' + data.name);
      }
      // ensure the json data is a string
      if (data.settings_name && !(typeof data.settings_name === 'string' || data.settings_name instanceof String)) {
        throw new Error('Expected the field `settings_name` to be a primitive type in the JSON string but got ' + data.settings_name);
      }
      // ensure the json data is a string
      if (data.iso && !(typeof data.iso === 'string' || data.iso instanceof String)) {
        throw new Error('Expected the field `iso` to be a primitive type in the JSON string but got ' + data.iso);
      }
      // ensure the json data is a string
      if (data.fiscal_iso && !(typeof data.fiscal_iso === 'string' || data.fiscal_iso instanceof String)) {
        throw new Error('Expected the field `fiscal_iso` to be a primitive type in the JSON string but got ' + data.fiscal_iso);
      }
      // ensure the json data is a string
      if (data.uic && !(typeof data.uic === 'string' || data.uic instanceof String)) {
        throw new Error('Expected the field `uic` to be a primitive type in the JSON string but got ' + data.uic);
      }
      return true;
    }
  }]);
  return DetailedCountry;
}();
/**
 * Country name
 * @member {String} name
 */
DetailedCountry.prototype.name = undefined;

/**
 * Country settings name
 * @member {String} settings_name
 */
DetailedCountry.prototype.settings_name = undefined;

/**
 * Country iso code
 * @member {String} iso
 */
DetailedCountry.prototype.iso = undefined;

/**
 * Country fiscal iso code
 * @member {String} fiscal_iso
 */
DetailedCountry.prototype.fiscal_iso = undefined;

/**
 * Country uic
 * @member {String} uic
 */
DetailedCountry.prototype.uic = undefined;
var _default = exports["default"] = DetailedCountry;
