// @fattureincloud/fattureincloud-js-sdk/dist/model/EInvoiceRejectionReason.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The EInvoiceRejectionReason model module.
 * @module model/EInvoiceRejectionReason
 * @version 2.0.25
 */
var EInvoiceRejectionReason = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>EInvoiceRejectionReason</code>.
     * @alias module:model/EInvoiceRejectionReason
     */
  function EInvoiceRejectionReason() {
    _classCallCheck(this, EInvoiceRejectionReason);
    EInvoiceRejectionReason.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(EInvoiceRejectionReason, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>EInvoiceRejectionReason</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/EInvoiceRejectionReason} obj Optional instance to populate.
       * @return {module:model/EInvoiceRejectionReason} The populated <code>EInvoiceRejectionReason</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EInvoiceRejectionReason();
        if (data.hasOwnProperty('reason')) {
          obj.reason = _ApiClient["default"].convertToType(data.reason, 'String');
        }
        if (data.hasOwnProperty('ei_status')) {
          obj.ei_status = _ApiClient["default"].convertToType(data.ei_status, 'String');
        }
        if (data.hasOwnProperty('solution')) {
          obj.solution = _ApiClient["default"].convertToType(data.solution, 'String');
        }
        if (data.hasOwnProperty('code')) {
          obj.code = _ApiClient["default"].convertToType(data.code, 'String');
        }
        if (data.hasOwnProperty('date')) {
          obj.date = _ApiClient["default"].convertToType(data.date, 'Date');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>EInvoiceRejectionReason</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>EInvoiceRejectionReason</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.reason && !(typeof data.reason === 'string' || data.reason instanceof String)) {
        throw new Error('Expected the field `reason` to be a primitive type in the JSON string but got ' + data.reason);
      }
      // ensure the json data is a string
      if (data.ei_status && !(typeof data.ei_status === 'string' || data.ei_status instanceof String)) {
        throw new Error('Expected the field `ei_status` to be a primitive type in the JSON string but got ' + data.ei_status);
      }
      // ensure the json data is a string
      if (data.solution && !(typeof data.solution === 'string' || data.solution instanceof String)) {
        throw new Error('Expected the field `solution` to be a primitive type in the JSON string but got ' + data.solution);
      }
      // ensure the json data is a string
      if (data.code && !(typeof data.code === 'string' || data.code instanceof String)) {
        throw new Error('Expected the field `code` to be a primitive type in the JSON string but got ' + data.code);
      }
      return true;
    }
  }]);
  return EInvoiceRejectionReason;
}();
/**
 * E-invoice rejection reason
 * @member {String} reason
 */
EInvoiceRejectionReason.prototype.reason = undefined;

/**
 * E-invoice status
 * @member {String} ei_status
 */
EInvoiceRejectionReason.prototype.ei_status = undefined;

/**
 * Error solution.
 * @member {String} solution
 */
EInvoiceRejectionReason.prototype.solution = undefined;

/**
 * E-invoice rejection error code
 * @member {String} code
 */
EInvoiceRejectionReason.prototype.code = undefined;

/**
 * E-invoice rejection date
 * @member {Date} date
 */
EInvoiceRejectionReason.prototype.date = undefined;
var _default = exports["default"] = EInvoiceRejectionReason;
