// @fattureincloud/fattureincloud-js-sdk/dist/model/EmailSchedule.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _EmailScheduleInclude = _interopRequireDefault(require("./EmailScheduleInclude"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The EmailSchedule model module.
 * @module model/EmailSchedule
 * @version 2.0.25
 */
var EmailSchedule = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>EmailSchedule</code>.
     * @alias module:model/EmailSchedule
     */
  function EmailSchedule() {
    _classCallCheck(this, EmailSchedule);
    EmailSchedule.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(EmailSchedule, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>EmailSchedule</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/EmailSchedule} obj Optional instance to populate.
       * @return {module:model/EmailSchedule} The populated <code>EmailSchedule</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EmailSchedule();
        if (data.hasOwnProperty('sender_id')) {
          obj.sender_id = _ApiClient["default"].convertToType(data.sender_id, 'Number');
        }
        if (data.hasOwnProperty('sender_email')) {
          obj.sender_email = _ApiClient["default"].convertToType(data.sender_email, 'String');
        }
        if (data.hasOwnProperty('recipient_email')) {
          obj.recipient_email = _ApiClient["default"].convertToType(data.recipient_email, 'String');
        }
        if (data.hasOwnProperty('subject')) {
          obj.subject = _ApiClient["default"].convertToType(data.subject, 'String');
        }
        if (data.hasOwnProperty('body')) {
          obj.body = _ApiClient["default"].convertToType(data.body, 'String');
        }
        if (data.hasOwnProperty('include')) {
          obj.include = _EmailScheduleInclude["default"].constructFromObject(data.include);
        }
        if (data.hasOwnProperty('attach_pdf')) {
          obj.attach_pdf = _ApiClient["default"].convertToType(data.attach_pdf, 'Boolean');
        }
        if (data.hasOwnProperty('send_copy')) {
          obj.send_copy = _ApiClient["default"].convertToType(data.send_copy, 'Boolean');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>EmailSchedule</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>EmailSchedule</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.sender_email && !(typeof data.sender_email === 'string' || data.sender_email instanceof String)) {
        throw new Error('Expected the field `sender_email` to be a primitive type in the JSON string but got ' + data.sender_email);
      }
      // ensure the json data is a string
      if (data.recipient_email && !(typeof data.recipient_email === 'string' || data.recipient_email instanceof String)) {
        throw new Error('Expected the field `recipient_email` to be a primitive type in the JSON string but got ' + data.recipient_email);
      }
      // ensure the json data is a string
      if (data.subject && !(typeof data.subject === 'string' || data.subject instanceof String)) {
        throw new Error('Expected the field `subject` to be a primitive type in the JSON string but got ' + data.subject);
      }
      // ensure the json data is a string
      if (data.body && !(typeof data.body === 'string' || data.body instanceof String)) {
        throw new Error('Expected the field `body` to be a primitive type in the JSON string but got ' + data.body);
      }
      // validate the optional field `include`
      if (data.include) {
        // data not null
        _EmailScheduleInclude["default"].validateJSON(data.include);
      }
      return true;
    }
  }]);
  return EmailSchedule;
}();
/**
 * Email sender id [required if **sender_email** is not specified]
 * @member {Number} sender_id
 */
EmailSchedule.prototype.sender_id = undefined;

/**
 * Email sender address [required if **sender_id** is not specified]
 * @member {String} sender_email
 */
EmailSchedule.prototype.sender_email = undefined;

/**
 * Email recipient emails [comma separated]
 * @member {String} recipient_email
 */
EmailSchedule.prototype.recipient_email = undefined;

/**
 * Email subject
 * @member {String} subject
 */
EmailSchedule.prototype.subject = undefined;

/**
 * Email body [HTML Escaped] [max size 50KiB]
 * @member {String} body
 */
EmailSchedule.prototype.body = undefined;

/**
 * @member {module:model/EmailScheduleInclude} include
 */
EmailSchedule.prototype.include = undefined;

/**
 * Attach the pdf of the document
 * @member {Boolean} attach_pdf
 */
EmailSchedule.prototype.attach_pdf = undefined;

/**
 * Send a copy of the email to the **cc_email** specified by **Get email data**
 * @member {Boolean} send_copy
 */
EmailSchedule.prototype.send_copy = undefined;
var _default = exports["default"] = EmailSchedule;
