// @fattureincloud/fattureincloud-js-sdk/dist/model/EmailData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _EmailDataDefaultSenderEmail = _interopRequireDefault(require("./EmailDataDefaultSenderEmail"));
var _SenderEmail = _interopRequireDefault(require("./SenderEmail"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The EmailData model module.
 * @module model/EmailData
 * @version 2.0.25
 */
var EmailData = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>EmailData</code>.
     * @alias module:model/EmailData
     */
  function EmailData() {
    _classCallCheck(this, EmailData);
    EmailData.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(EmailData, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>EmailData</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/EmailData} obj Optional instance to populate.
       * @return {module:model/EmailData} The populated <code>EmailData</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EmailData();
        if (data.hasOwnProperty('recipient_email')) {
          obj.recipient_email = _ApiClient["default"].convertToType(data.recipient_email, 'String');
        }
        if (data.hasOwnProperty('default_sender_email')) {
          obj.default_sender_email = _EmailDataDefaultSenderEmail["default"].constructFromObject(data.default_sender_email);
        }
        if (data.hasOwnProperty('sender_emails_list')) {
          obj.sender_emails_list = _ApiClient["default"].convertToType(data.sender_emails_list, [_SenderEmail["default"]]);
        }
        if (data.hasOwnProperty('cc_email')) {
          obj.cc_email = _ApiClient["default"].convertToType(data.cc_email, 'String');
        }
        if (data.hasOwnProperty('subject')) {
          obj.subject = _ApiClient["default"].convertToType(data.subject, 'String');
        }
        if (data.hasOwnProperty('body')) {
          obj.body = _ApiClient["default"].convertToType(data.body, 'String');
        }
        if (data.hasOwnProperty('document_exists')) {
          obj.document_exists = _ApiClient["default"].convertToType(data.document_exists, 'Boolean');
        }
        if (data.hasOwnProperty('delivery_note_exists')) {
          obj.delivery_note_exists = _ApiClient["default"].convertToType(data.delivery_note_exists, 'Boolean');
        }
        if (data.hasOwnProperty('attachment_exists')) {
          obj.attachment_exists = _ApiClient["default"].convertToType(data.attachment_exists, 'Boolean');
        }
        if (data.hasOwnProperty('accompanying_invoice_exists')) {
          obj.accompanying_invoice_exists = _ApiClient["default"].convertToType(data.accompanying_invoice_exists, 'Boolean');
        }
        if (data.hasOwnProperty('default_attach_pdf')) {
          obj.default_attach_pdf = _ApiClient["default"].convertToType(data.default_attach_pdf, 'Boolean');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>EmailData</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>EmailData</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.recipient_email && !(typeof data.recipient_email === 'string' || data.recipient_email instanceof String)) {
        throw new Error('Expected the field `recipient_email` to be a primitive type in the JSON string but got ' + data.recipient_email);
      }
      // validate the optional field `default_sender_email`
      if (data.default_sender_email) {
        // data not null
        _EmailDataDefaultSenderEmail["default"].validateJSON(data.default_sender_email);
      }
      if (data.sender_emails_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.sender_emails_list)) {
          throw new Error('Expected the field `sender_emails_list` to be an array in the JSON data but got ' + data.sender_emails_list);
        }
        // validate the optional field `sender_emails_list` (array)
        var _iterator = _createForOfIteratorHelper(data.sender_emails_list),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _SenderEmail["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
      }
      // ensure the json data is a string
      if (data.cc_email && !(typeof data.cc_email === 'string' || data.cc_email instanceof String)) {
        throw new Error('Expected the field `cc_email` to be a primitive type in the JSON string but got ' + data.cc_email);
      }
      // ensure the json data is a string
      if (data.subject && !(typeof data.subject === 'string' || data.subject instanceof String)) {
        throw new Error('Expected the field `subject` to be a primitive type in the JSON string but got ' + data.subject);
      }
      // ensure the json data is a string
      if (data.body && !(typeof data.body === 'string' || data.body instanceof String)) {
        throw new Error('Expected the field `body` to be a primitive type in the JSON string but got ' + data.body);
      }
      return true;
    }
  }]);
  return EmailData;
}();
/**
 * Email recipient
 * @member {String} recipient_email
 */
EmailData.prototype.recipient_email = undefined;

/**
 * @member {module:model/EmailDataDefaultSenderEmail} default_sender_email
 */
EmailData.prototype.default_sender_email = undefined;

/**
 * List of all emails from which the document can be sent
 * @member {Array.<module:model/SenderEmail>} sender_emails_list
 */
EmailData.prototype.sender_emails_list = undefined;

/**
 * Email cc [by default is the logged company email]
 * @member {String} cc_email
 */
EmailData.prototype.cc_email = undefined;

/**
 * Email subject
 * @member {String} subject
 */
EmailData.prototype.subject = undefined;

/**
 * Email body
 * @member {String} body
 */
EmailData.prototype.body = undefined;

/**
 * Document exists if it is not a delivery note
 * @member {Boolean} document_exists
 */
EmailData.prototype.document_exists = undefined;

/**
 * Document is a delivery note
 * @member {Boolean} delivery_note_exists
 */
EmailData.prototype.delivery_note_exists = undefined;

/**
 * Document has attachment
 * @member {Boolean} attachment_exists
 */
EmailData.prototype.attachment_exists = undefined;

/**
 * Document has accompanying invoice
 * @member {Boolean} accompanying_invoice_exists
 */
EmailData.prototype.accompanying_invoice_exists = undefined;

/**
 * Attach document pdf
 * @member {Boolean} default_attach_pdf
 */
EmailData.prototype.default_attach_pdf = undefined;
var _default = exports["default"] = EmailData;
