// @fattureincloud/fattureincloud-js-sdk/dist/model/F24.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _F24Status = _interopRequireDefault(require("./F24Status"));
var _PaymentAccount = _interopRequireDefault(require("./PaymentAccount"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The F24 model module.
 * @module model/F24
 * @version 2.0.25
 */
var F24 = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>F24</code>.
     * @alias module:model/F24
     */
  function F24() {
    _classCallCheck(this, F24);
    F24.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(F24, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>F24</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/F24} obj Optional instance to populate.
       * @return {module:model/F24} The populated <code>F24</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new F24();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'Number');
        }
        if (data.hasOwnProperty('due_date')) {
          obj.due_date = _ApiClient["default"].convertToType(data.due_date, 'Date');
        }
        if (data.hasOwnProperty('status')) {
          obj.status = _F24Status["default"].constructFromObject(data.status);
        }
        if (data.hasOwnProperty('payment_account')) {
          obj.payment_account = _PaymentAccount["default"].constructFromObject(data.payment_account);
        }
        if (data.hasOwnProperty('amount')) {
          obj.amount = _ApiClient["default"].convertToType(data.amount, 'Number');
        }
        if (data.hasOwnProperty('attachment_url')) {
          obj.attachment_url = _ApiClient["default"].convertToType(data.attachment_url, 'String');
        }
        if (data.hasOwnProperty('attachment_token')) {
          obj.attachment_token = _ApiClient["default"].convertToType(data.attachment_token, 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj.description = _ApiClient["default"].convertToType(data.description, 'String');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>F24</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>F24</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `payment_account`
      if (data.payment_account) {
        // data not null
        _PaymentAccount["default"].validateJSON(data.payment_account);
      }
      // ensure the json data is a string
      if (data.attachment_url && !(typeof data.attachment_url === 'string' || data.attachment_url instanceof String)) {
        throw new Error('Expected the field `attachment_url` to be a primitive type in the JSON string but got ' + data.attachment_url);
      }
      // ensure the json data is a string
      if (data.attachment_token && !(typeof data.attachment_token === 'string' || data.attachment_token instanceof String)) {
        throw new Error('Expected the field `attachment_token` to be a primitive type in the JSON string but got ' + data.attachment_token);
      }
      // ensure the json data is a string
      if (data.description && !(typeof data.description === 'string' || data.description instanceof String)) {
        throw new Error('Expected the field `description` to be a primitive type in the JSON string but got ' + data.description);
      }
      return true;
    }
  }]);
  return F24;
}();
/**
 * F24 id
 * @member {Number} id
 */
F24.prototype.id = undefined;

/**
 * F24 due date
 * @member {Date} due_date
 */
F24.prototype.due_date = undefined;

/**
 * @member {module:model/F24Status} status
 */
F24.prototype.status = undefined;

/**
 * @member {module:model/PaymentAccount} payment_account
 */
F24.prototype.payment_account = undefined;

/**
 * F24 amount
 * @member {Number} amount
 */
F24.prototype.amount = undefined;

/**
 * [Temporary] [Read Only] F24 url of the attached file
 * @member {String} attachment_url
 */
F24.prototype.attachment_url = undefined;

/**
 * [Write Only]  F24 attachment token returned by POST /taxes/attachment
 * @member {String} attachment_token
 */
F24.prototype.attachment_token = undefined;

/**
 * F24 description
 * @member {String} description
 */
F24.prototype.description = undefined;
var _default = exports["default"] = F24;
