// @fattureincloud/fattureincloud-js-sdk/dist/model/IssuedDocumentOptions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The IssuedDocumentOptions model module.
 * @module model/IssuedDocumentOptions
 * @version 2.0.25
 */
var IssuedDocumentOptions = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>IssuedDocumentOptions</code>.
     * @alias module:model/IssuedDocumentOptions
     */
  function IssuedDocumentOptions() {
    _classCallCheck(this, IssuedDocumentOptions);
    IssuedDocumentOptions.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(IssuedDocumentOptions, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>IssuedDocumentOptions</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/IssuedDocumentOptions} obj Optional instance to populate.
       * @return {module:model/IssuedDocumentOptions} The populated <code>IssuedDocumentOptions</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IssuedDocumentOptions();
        if (data.hasOwnProperty('fix_payments')) {
          obj.fix_payments = _ApiClient["default"].convertToType(data.fix_payments, 'Boolean');
        }
        if (data.hasOwnProperty('create_from')) {
          obj.create_from = _ApiClient["default"].convertToType(data.create_from, ['String']);
        }
        if (data.hasOwnProperty('transform')) {
          obj.transform = _ApiClient["default"].convertToType(data.transform, 'Boolean');
        }
        if (data.hasOwnProperty('keep_copy')) {
          obj.keep_copy = _ApiClient["default"].convertToType(data.keep_copy, 'Boolean');
        }
        if (data.hasOwnProperty('join_type')) {
          obj.join_type = _ApiClient["default"].convertToType(data.join_type, 'String');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>IssuedDocumentOptions</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>IssuedDocumentOptions</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is an array
      if (!Array.isArray(data.create_from)) {
        throw new Error('Expected the field `create_from` to be an array in the JSON data but got ' + data.create_from);
      }
      // ensure the json data is a string
      if (data.join_type && !(typeof data.join_type === 'string' || data.join_type instanceof String)) {
        throw new Error('Expected the field `join_type` to be a primitive type in the JSON string but got ' + data.join_type);
      }
      return true;
    }
  }]);
  return IssuedDocumentOptions;
}();
/**
 * Fixes your last payment amount to match your document total
 * @member {Boolean} fix_payments
 */
IssuedDocumentOptions.prototype.fix_payments = undefined;

/**
 * Original documents ids [only for join/transform]
 * @member {Array.<String>} create_from
 */
IssuedDocumentOptions.prototype.create_from = undefined;

/**
 * Tranform a document [only for transform]
 * @member {Boolean} transform
 */
IssuedDocumentOptions.prototype.transform = undefined;

/**
 * Keep original document [only for transform]
 * @member {Boolean} keep_copy
 */
IssuedDocumentOptions.prototype.keep_copy = undefined;

/**
 * Join type [only for join]
 * @member {String} join_type
 */
IssuedDocumentOptions.prototype.join_type = undefined;
var _default = exports["default"] = IssuedDocumentOptions;
