// @fattureincloud/fattureincloud-js-sdk/dist/model/IssuedDocumentPaymentsListItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _IssuedDocumentPaymentsListItemPaymentTerms = _interopRequireDefault(require("./IssuedDocumentPaymentsListItemPaymentTerms"));
var _IssuedDocumentStatus = _interopRequireDefault(require("./IssuedDocumentStatus"));
var _PaymentAccount = _interopRequireDefault(require("./PaymentAccount"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The IssuedDocumentPaymentsListItem model module.
 * @module model/IssuedDocumentPaymentsListItem
 * @version 2.0.25
 */
var IssuedDocumentPaymentsListItem = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>IssuedDocumentPaymentsListItem</code>.
     * @alias module:model/IssuedDocumentPaymentsListItem
     */
  function IssuedDocumentPaymentsListItem() {
    _classCallCheck(this, IssuedDocumentPaymentsListItem);
    IssuedDocumentPaymentsListItem.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(IssuedDocumentPaymentsListItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>IssuedDocumentPaymentsListItem</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/IssuedDocumentPaymentsListItem} obj Optional instance to populate.
       * @return {module:model/IssuedDocumentPaymentsListItem} The populated <code>IssuedDocumentPaymentsListItem</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IssuedDocumentPaymentsListItem();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'Number');
        }
        if (data.hasOwnProperty('due_date')) {
          obj.due_date = _ApiClient["default"].convertToType(data.due_date, 'Date');
        }
        if (data.hasOwnProperty('amount')) {
          obj.amount = _ApiClient["default"].convertToType(data.amount, 'Number');
        }
        if (data.hasOwnProperty('status')) {
          obj.status = _IssuedDocumentStatus["default"].constructFromObject(data.status);
        }
        if (data.hasOwnProperty('payment_account')) {
          obj.payment_account = _PaymentAccount["default"].constructFromObject(data.payment_account);
        }
        if (data.hasOwnProperty('paid_date')) {
          obj.paid_date = _ApiClient["default"].convertToType(data.paid_date, 'Date');
        }
        if (data.hasOwnProperty('ei_raw')) {
          obj.ei_raw = _ApiClient["default"].convertToType(data.ei_raw, Object);
        }
        if (data.hasOwnProperty('payment_terms')) {
          obj.payment_terms = _IssuedDocumentPaymentsListItemPaymentTerms["default"].constructFromObject(data.payment_terms);
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>IssuedDocumentPaymentsListItem</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>IssuedDocumentPaymentsListItem</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `payment_account`
      if (data.payment_account) {
        // data not null
        _PaymentAccount["default"].validateJSON(data.payment_account);
      }
      // validate the optional field `payment_terms`
      if (data.payment_terms) {
        // data not null
        _IssuedDocumentPaymentsListItemPaymentTerms["default"].validateJSON(data.payment_terms);
      }
      return true;
    }
  }]);
  return IssuedDocumentPaymentsListItem;
}();
/**
 * Issued document payment item id
 * @member {Number} id
 */
IssuedDocumentPaymentsListItem.prototype.id = undefined;

/**
 * Issued document payment due date
 * @member {Date} due_date
 */
IssuedDocumentPaymentsListItem.prototype.due_date = undefined;

/**
 * Issued document payment amount
 * @member {Number} amount
 */
IssuedDocumentPaymentsListItem.prototype.amount = undefined;

/**
 * @member {module:model/IssuedDocumentStatus} status
 */
IssuedDocumentPaymentsListItem.prototype.status = undefined;

/**
 * @member {module:model/PaymentAccount} payment_account
 */
IssuedDocumentPaymentsListItem.prototype.payment_account = undefined;

/**
 * Issued document payment date [Only if status is paid]
 * @member {Date} paid_date
 */
IssuedDocumentPaymentsListItem.prototype.paid_date = undefined;

/**
 * Issued document payment advanced raw attributes for e-invoices
 * @member {Object} ei_raw
 */
IssuedDocumentPaymentsListItem.prototype.ei_raw = undefined;

/**
 * @member {module:model/IssuedDocumentPaymentsListItemPaymentTerms} payment_terms
 */
IssuedDocumentPaymentsListItem.prototype.payment_terms = undefined;
var _default = exports["default"] = IssuedDocumentPaymentsListItem;
