// @fattureincloud/fattureincloud-js-sdk/dist/model/IssuedDocumentExtraData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The IssuedDocumentExtraData model module.
 * @module model/IssuedDocumentExtraData
 * @version 2.0.25
 */
var IssuedDocumentExtraData = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>IssuedDocumentExtraData</code>.
     * Issued document extra data [TS fields follow the technical specifications provided by \&quot;Sistema Tessera Sanitaria\&quot;]
     * @alias module:model/IssuedDocumentExtraData
     */
  function IssuedDocumentExtraData() {
    _classCallCheck(this, IssuedDocumentExtraData);
    IssuedDocumentExtraData.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(IssuedDocumentExtraData, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>IssuedDocumentExtraData</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/IssuedDocumentExtraData} obj Optional instance to populate.
       * @return {module:model/IssuedDocumentExtraData} The populated <code>IssuedDocumentExtraData</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IssuedDocumentExtraData();
        if (data.hasOwnProperty('show_sofort_button')) {
          obj.show_sofort_button = _ApiClient["default"].convertToType(data.show_sofort_button, 'Boolean');
        }
        if (data.hasOwnProperty('multifatture_sent')) {
          obj.multifatture_sent = _ApiClient["default"].convertToType(data.multifatture_sent, 'Number');
        }
        if (data.hasOwnProperty('ts_communication')) {
          obj.ts_communication = _ApiClient["default"].convertToType(data.ts_communication, 'Boolean');
        }
        if (data.hasOwnProperty('ts_flag_tipo_spesa')) {
          obj.ts_flag_tipo_spesa = _ApiClient["default"].convertToType(data.ts_flag_tipo_spesa, 'Number');
        }
        if (data.hasOwnProperty('ts_pagamento_tracciato')) {
          obj.ts_pagamento_tracciato = _ApiClient["default"].convertToType(data.ts_pagamento_tracciato, 'Boolean');
        }
        if (data.hasOwnProperty('ts_tipo_spesa')) {
          obj.ts_tipo_spesa = _ApiClient["default"].convertToType(data.ts_tipo_spesa, 'String');
        }
        if (data.hasOwnProperty('ts_opposizione')) {
          obj.ts_opposizione = _ApiClient["default"].convertToType(data.ts_opposizione, 'Boolean');
        }
        if (data.hasOwnProperty('ts_status')) {
          obj.ts_status = _ApiClient["default"].convertToType(data.ts_status, 'Number');
        }
        if (data.hasOwnProperty('ts_file_id')) {
          obj.ts_file_id = _ApiClient["default"].convertToType(data.ts_file_id, 'String');
        }
        if (data.hasOwnProperty('ts_sent_date')) {
          obj.ts_sent_date = _ApiClient["default"].convertToType(data.ts_sent_date, 'Date');
        }
        if (data.hasOwnProperty('ts_full_amount')) {
          obj.ts_full_amount = _ApiClient["default"].convertToType(data.ts_full_amount, 'Boolean');
        }
        if (data.hasOwnProperty('imported_by')) {
          obj.imported_by = _ApiClient["default"].convertToType(data.imported_by, 'String');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>IssuedDocumentExtraData</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>IssuedDocumentExtraData</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.ts_tipo_spesa && !(typeof data.ts_tipo_spesa === 'string' || data.ts_tipo_spesa instanceof String)) {
        throw new Error('Expected the field `ts_tipo_spesa` to be a primitive type in the JSON string but got ' + data.ts_tipo_spesa);
      }
      // ensure the json data is a string
      if (data.ts_file_id && !(typeof data.ts_file_id === 'string' || data.ts_file_id instanceof String)) {
        throw new Error('Expected the field `ts_file_id` to be a primitive type in the JSON string but got ' + data.ts_file_id);
      }
      // ensure the json data is a string
      if (data.imported_by && !(typeof data.imported_by === 'string' || data.imported_by instanceof String)) {
        throw new Error('Expected the field `imported_by` to be a primitive type in the JSON string but got ' + data.imported_by);
      }
      return true;
    }
  }]);
  return IssuedDocumentExtraData;
}();
/**
 * @member {Boolean} show_sofort_button
 */
IssuedDocumentExtraData.prototype.show_sofort_button = undefined;

/**
 * @member {Number} multifatture_sent
 */
IssuedDocumentExtraData.prototype.multifatture_sent = undefined;

/**
 * Send issued document to \"Sistema Tessera Sanitaria\"
 * @member {Boolean} ts_communication
 */
IssuedDocumentExtraData.prototype.ts_communication = undefined;

/**
 * Issued document ts \"tipo spesa\" [TK, FC, FV, SV,SP, AD, AS, ECG, SR]
 * @member {Number} ts_flag_tipo_spesa
 */
IssuedDocumentExtraData.prototype.ts_flag_tipo_spesa = undefined;

/**
 * Issued document ts traced payment
 * @member {Boolean} ts_pagamento_tracciato
 */
IssuedDocumentExtraData.prototype.ts_pagamento_tracciato = undefined;

/**
 * Can be [ 'TK', 'FC', 'FV', 'SV', 'SP', 'AD', 'AS', 'SR', 'CT', 'PI', 'IC', 'AA' ]. Refer to the technical specifications to learn more.
 * @member {String} ts_tipo_spesa
 */
IssuedDocumentExtraData.prototype.ts_tipo_spesa = undefined;

/**
 * Issued document ts \"opposizione\"
 * @member {Boolean} ts_opposizione
 */
IssuedDocumentExtraData.prototype.ts_opposizione = undefined;

/**
 * Issued document ts status
 * @member {Number} ts_status
 */
IssuedDocumentExtraData.prototype.ts_status = undefined;

/**
 * Issued document ts file id
 * @member {String} ts_file_id
 */
IssuedDocumentExtraData.prototype.ts_file_id = undefined;

/**
 * Issued document ts sent date
 * @member {Date} ts_sent_date
 */
IssuedDocumentExtraData.prototype.ts_sent_date = undefined;

/**
 * Issued document ts total amount
 * @member {Boolean} ts_full_amount
 */
IssuedDocumentExtraData.prototype.ts_full_amount = undefined;

/**
 * Issued document imported by software
 * @member {String} imported_by
 */
IssuedDocumentExtraData.prototype.imported_by = undefined;
var _default = exports["default"] = IssuedDocumentExtraData;
