// @fattureincloud/fattureincloud-js-sdk/dist/model/IssuedDocumentPreCreateInfoDefaultValues.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _DocumentTemplate = _interopRequireDefault(require("./DocumentTemplate"));
var _PaymentMethod = _interopRequireDefault(require("./PaymentMethod"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The IssuedDocumentPreCreateInfoDefaultValues model module.
 * @module model/IssuedDocumentPreCreateInfoDefaultValues
 * @version 2.0.25
 */
var IssuedDocumentPreCreateInfoDefaultValues = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>IssuedDocumentPreCreateInfoDefaultValues</code>.
     * Issued document default values
     * @alias module:model/IssuedDocumentPreCreateInfoDefaultValues
     */
  function IssuedDocumentPreCreateInfoDefaultValues() {
    _classCallCheck(this, IssuedDocumentPreCreateInfoDefaultValues);
    IssuedDocumentPreCreateInfoDefaultValues.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(IssuedDocumentPreCreateInfoDefaultValues, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>IssuedDocumentPreCreateInfoDefaultValues</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/IssuedDocumentPreCreateInfoDefaultValues} obj Optional instance to populate.
       * @return {module:model/IssuedDocumentPreCreateInfoDefaultValues} The populated <code>IssuedDocumentPreCreateInfoDefaultValues</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IssuedDocumentPreCreateInfoDefaultValues();
        if (data.hasOwnProperty('default_template')) {
          obj.default_template = _DocumentTemplate["default"].constructFromObject(data.default_template);
        }
        if (data.hasOwnProperty('dn_template')) {
          obj.dn_template = _DocumentTemplate["default"].constructFromObject(data.dn_template);
        }
        if (data.hasOwnProperty('ai_template')) {
          obj.ai_template = _DocumentTemplate["default"].constructFromObject(data.ai_template);
        }
        if (data.hasOwnProperty('notes')) {
          obj.notes = _ApiClient["default"].convertToType(data.notes, 'String');
        }
        if (data.hasOwnProperty('rivalsa')) {
          obj.rivalsa = _ApiClient["default"].convertToType(data.rivalsa, 'Number');
        }
        if (data.hasOwnProperty('cassa')) {
          obj.cassa = _ApiClient["default"].convertToType(data.cassa, 'Number');
        }
        if (data.hasOwnProperty('withholding_tax')) {
          obj.withholding_tax = _ApiClient["default"].convertToType(data.withholding_tax, 'Number');
        }
        if (data.hasOwnProperty('withholding_tax_taxable')) {
          obj.withholding_tax_taxable = _ApiClient["default"].convertToType(data.withholding_tax_taxable, 'Number');
        }
        if (data.hasOwnProperty('other_withholding_tax')) {
          obj.other_withholding_tax = _ApiClient["default"].convertToType(data.other_withholding_tax, 'Number');
        }
        if (data.hasOwnProperty('use_gross_prices')) {
          obj.use_gross_prices = _ApiClient["default"].convertToType(data.use_gross_prices, 'Boolean');
        }
        if (data.hasOwnProperty('payment_method')) {
          obj.payment_method = _PaymentMethod["default"].constructFromObject(data.payment_method);
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>IssuedDocumentPreCreateInfoDefaultValues</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>IssuedDocumentPreCreateInfoDefaultValues</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `default_template`
      if (data.default_template) {
        // data not null
        _DocumentTemplate["default"].validateJSON(data.default_template);
      }
      // validate the optional field `dn_template`
      if (data.dn_template) {
        // data not null
        _DocumentTemplate["default"].validateJSON(data.dn_template);
      }
      // validate the optional field `ai_template`
      if (data.ai_template) {
        // data not null
        _DocumentTemplate["default"].validateJSON(data.ai_template);
      }
      // ensure the json data is a string
      if (data.notes && !(typeof data.notes === 'string' || data.notes instanceof String)) {
        throw new Error('Expected the field `notes` to be a primitive type in the JSON string but got ' + data.notes);
      }
      // validate the optional field `payment_method`
      if (data.payment_method) {
        // data not null
        _PaymentMethod["default"].validateJSON(data.payment_method);
      }
      return true;
    }
  }]);
  return IssuedDocumentPreCreateInfoDefaultValues;
}();
/**
 * @member {module:model/DocumentTemplate} default_template
 */
IssuedDocumentPreCreateInfoDefaultValues.prototype.default_template = undefined;

/**
 * @member {module:model/DocumentTemplate} dn_template
 */
IssuedDocumentPreCreateInfoDefaultValues.prototype.dn_template = undefined;

/**
 * @member {module:model/DocumentTemplate} ai_template
 */
IssuedDocumentPreCreateInfoDefaultValues.prototype.ai_template = undefined;

/**
 * Default notes.
 * @member {String} notes
 */
IssuedDocumentPreCreateInfoDefaultValues.prototype.notes = undefined;

/**
 * Default rivalsa percentage.
 * @member {Number} rivalsa
 */
IssuedDocumentPreCreateInfoDefaultValues.prototype.rivalsa = undefined;

/**
 * Default cassa percentage.
 * @member {Number} cassa
 */
IssuedDocumentPreCreateInfoDefaultValues.prototype.cassa = undefined;

/**
 * Default withholding tax percentage.
 * @member {Number} withholding_tax
 */
IssuedDocumentPreCreateInfoDefaultValues.prototype.withholding_tax = undefined;

/**
 * Default withholding tax taxable percentage.
 * @member {Number} withholding_tax_taxable
 */
IssuedDocumentPreCreateInfoDefaultValues.prototype.withholding_tax_taxable = undefined;

/**
 * Default other withholding tax percentage.
 * @member {Number} other_withholding_tax
 */
IssuedDocumentPreCreateInfoDefaultValues.prototype.other_withholding_tax = undefined;

/**
 * Use gross price by default.
 * @member {Boolean} use_gross_prices
 */
IssuedDocumentPreCreateInfoDefaultValues.prototype.use_gross_prices = undefined;

/**
 * @member {module:model/PaymentMethod} payment_method
 */
IssuedDocumentPreCreateInfoDefaultValues.prototype.payment_method = undefined;
var _default = exports["default"] = IssuedDocumentPreCreateInfoDefaultValues;
