// @fattureincloud/fattureincloud-js-sdk/dist/model/IssuedDocumentItemsListItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _VatType = _interopRequireDefault(require("./VatType"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The IssuedDocumentItemsListItem model module.
 * @module model/IssuedDocumentItemsListItem
 * @version 2.0.25
 */
var IssuedDocumentItemsListItem = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>IssuedDocumentItemsListItem</code>.
     * @alias module:model/IssuedDocumentItemsListItem
     */
  function IssuedDocumentItemsListItem() {
    _classCallCheck(this, IssuedDocumentItemsListItem);
    IssuedDocumentItemsListItem.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(IssuedDocumentItemsListItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>IssuedDocumentItemsListItem</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/IssuedDocumentItemsListItem} obj Optional instance to populate.
       * @return {module:model/IssuedDocumentItemsListItem} The populated <code>IssuedDocumentItemsListItem</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IssuedDocumentItemsListItem();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'Number');
        }
        if (data.hasOwnProperty('product_id')) {
          obj.product_id = _ApiClient["default"].convertToType(data.product_id, 'Number');
        }
        if (data.hasOwnProperty('code')) {
          obj.code = _ApiClient["default"].convertToType(data.code, 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj.name = _ApiClient["default"].convertToType(data.name, 'String');
        }
        if (data.hasOwnProperty('category')) {
          obj.category = _ApiClient["default"].convertToType(data.category, 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj.description = _ApiClient["default"].convertToType(data.description, 'String');
        }
        if (data.hasOwnProperty('qty')) {
          obj.qty = _ApiClient["default"].convertToType(data.qty, 'Number');
        }
        if (data.hasOwnProperty('measure')) {
          obj.measure = _ApiClient["default"].convertToType(data.measure, 'String');
        }
        if (data.hasOwnProperty('net_price')) {
          obj.net_price = _ApiClient["default"].convertToType(data.net_price, 'Number');
        }
        if (data.hasOwnProperty('gross_price')) {
          obj.gross_price = _ApiClient["default"].convertToType(data.gross_price, 'Number');
        }
        if (data.hasOwnProperty('vat')) {
          obj.vat = _VatType["default"].constructFromObject(data.vat);
        }
        if (data.hasOwnProperty('not_taxable')) {
          obj.not_taxable = _ApiClient["default"].convertToType(data.not_taxable, 'Boolean');
        }
        if (data.hasOwnProperty('apply_withholding_taxes')) {
          obj.apply_withholding_taxes = _ApiClient["default"].convertToType(data.apply_withholding_taxes, 'Boolean');
        }
        if (data.hasOwnProperty('discount')) {
          obj.discount = _ApiClient["default"].convertToType(data.discount, 'Number');
        }
        if (data.hasOwnProperty('discount_highlight')) {
          obj.discount_highlight = _ApiClient["default"].convertToType(data.discount_highlight, 'Boolean');
        }
        if (data.hasOwnProperty('in_dn')) {
          obj.in_dn = _ApiClient["default"].convertToType(data.in_dn, 'Boolean');
        }
        if (data.hasOwnProperty('stock')) {
          obj.stock = _ApiClient["default"].convertToType(data.stock, 'Boolean');
        }
        if (data.hasOwnProperty('ei_raw')) {
          obj.ei_raw = _ApiClient["default"].convertToType(data.ei_raw, Object);
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>IssuedDocumentItemsListItem</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>IssuedDocumentItemsListItem</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.code && !(typeof data.code === 'string' || data.code instanceof String)) {
        throw new Error('Expected the field `code` to be a primitive type in the JSON string but got ' + data.code);
      }
      // ensure the json data is a string
      if (data.name && !(typeof data.name === 'string' || data.name instanceof String)) {
        throw new Error('Expected the field `name` to be a primitive type in the JSON string but got ' + data.name);
      }
      // ensure the json data is a string
      if (data.category && !(typeof data.category === 'string' || data.category instanceof String)) {
        throw new Error('Expected the field `category` to be a primitive type in the JSON string but got ' + data.category);
      }
      // ensure the json data is a string
      if (data.description && !(typeof data.description === 'string' || data.description instanceof String)) {
        throw new Error('Expected the field `description` to be a primitive type in the JSON string but got ' + data.description);
      }
      // ensure the json data is a string
      if (data.measure && !(typeof data.measure === 'string' || data.measure instanceof String)) {
        throw new Error('Expected the field `measure` to be a primitive type in the JSON string but got ' + data.measure);
      }
      // validate the optional field `vat`
      if (data.vat) {
        // data not null
        _VatType["default"].validateJSON(data.vat);
      }
      return true;
    }
  }]);
  return IssuedDocumentItemsListItem;
}();
/**
 * Issued document item id
 * @member {Number} id
 */
IssuedDocumentItemsListItem.prototype.id = undefined;

/**
 * Issued document item product id
 * @member {Number} product_id
 */
IssuedDocumentItemsListItem.prototype.product_id = undefined;

/**
 * Issued document item product code
 * @member {String} code
 */
IssuedDocumentItemsListItem.prototype.code = undefined;

/**
 * Issued document item product name
 * @member {String} name
 */
IssuedDocumentItemsListItem.prototype.name = undefined;

/**
 * Issued document item product category
 * @member {String} category
 */
IssuedDocumentItemsListItem.prototype.category = undefined;

/**
 * Issued document product description
 * @member {String} description
 */
IssuedDocumentItemsListItem.prototype.description = undefined;

/**
 * Issued document item quantity
 * @member {Number} qty
 */
IssuedDocumentItemsListItem.prototype.qty = undefined;

/**
 * Issued document item measure
 * @member {String} measure
 */
IssuedDocumentItemsListItem.prototype.measure = undefined;

/**
 * Issued document item net price
 * @member {Number} net_price
 */
IssuedDocumentItemsListItem.prototype.net_price = undefined;

/**
 * Issued document item gross price
 * @member {Number} gross_price
 */
IssuedDocumentItemsListItem.prototype.gross_price = undefined;

/**
 * @member {module:model/VatType} vat
 */
IssuedDocumentItemsListItem.prototype.vat = undefined;

/**
 * Issued document item is not taxable
 * @member {Boolean} not_taxable
 */
IssuedDocumentItemsListItem.prototype.not_taxable = undefined;

/**
 * Issued document item apply withholding taxes, rivalsa and cassa
 * @member {Boolean} apply_withholding_taxes
 */
IssuedDocumentItemsListItem.prototype.apply_withholding_taxes = undefined;

/**
 * Issued document item discount percentual value
 * @member {Number} discount
 */
IssuedDocumentItemsListItem.prototype.discount = undefined;

/**
 * Issued document item highlight discount
 * @member {Boolean} discount_highlight
 */
IssuedDocumentItemsListItem.prototype.discount_highlight = undefined;

/**
 * Issued document item add in delivery note
 * @member {Boolean} in_dn
 */
IssuedDocumentItemsListItem.prototype.in_dn = undefined;

/**
 * Issued document item move stock
 * @member {Boolean} stock
 */
IssuedDocumentItemsListItem.prototype.stock = undefined;

/**
 * Issued document advanced raw attributes for e-invoices
 * @member {Object} ei_raw
 */
IssuedDocumentItemsListItem.prototype.ei_raw = undefined;
var _default = exports["default"] = IssuedDocumentItemsListItem;
