// @fattureincloud/fattureincloud-js-sdk/dist/model/IssuedDocumentPreCreateInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Currency = _interopRequireDefault(require("./Currency"));
var _DocumentTemplate = _interopRequireDefault(require("./DocumentTemplate"));
var _IssuedDocumentPreCreateInfoDefaultValues = _interopRequireDefault(require("./IssuedDocumentPreCreateInfoDefaultValues"));
var _IssuedDocumentPreCreateInfoExtraDataDefaultValues = _interopRequireDefault(require("./IssuedDocumentPreCreateInfoExtraDataDefaultValues"));
var _IssuedDocumentPreCreateInfoItemsDefaultValues = _interopRequireDefault(require("./IssuedDocumentPreCreateInfoItemsDefaultValues"));
var _Language = _interopRequireDefault(require("./Language"));
var _PaymentAccount = _interopRequireDefault(require("./PaymentAccount"));
var _PaymentMethod = _interopRequireDefault(require("./PaymentMethod"));
var _VatType = _interopRequireDefault(require("./VatType"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The IssuedDocumentPreCreateInfo model module.
 * @module model/IssuedDocumentPreCreateInfo
 * @version 2.0.25
 */
var IssuedDocumentPreCreateInfo = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>IssuedDocumentPreCreateInfo</code>.
     * @alias module:model/IssuedDocumentPreCreateInfo
     */
  function IssuedDocumentPreCreateInfo() {
    _classCallCheck(this, IssuedDocumentPreCreateInfo);
    IssuedDocumentPreCreateInfo.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(IssuedDocumentPreCreateInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>IssuedDocumentPreCreateInfo</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/IssuedDocumentPreCreateInfo} obj Optional instance to populate.
       * @return {module:model/IssuedDocumentPreCreateInfo} The populated <code>IssuedDocumentPreCreateInfo</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IssuedDocumentPreCreateInfo();
        if (data.hasOwnProperty('numerations')) {
          obj.numerations = _ApiClient["default"].convertToType(data.numerations, {
            String: Object
          });
        }
        if (data.hasOwnProperty('dn_numerations')) {
          obj.dn_numerations = _ApiClient["default"].convertToType(data.dn_numerations, {
            String: Object
          });
        }
        if (data.hasOwnProperty('default_values')) {
          obj.default_values = _IssuedDocumentPreCreateInfoDefaultValues["default"].constructFromObject(data.default_values);
        }
        if (data.hasOwnProperty('extra_data_default_values')) {
          obj.extra_data_default_values = _IssuedDocumentPreCreateInfoExtraDataDefaultValues["default"].constructFromObject(data.extra_data_default_values);
        }
        if (data.hasOwnProperty('items_default_values')) {
          obj.items_default_values = _IssuedDocumentPreCreateInfoItemsDefaultValues["default"].constructFromObject(data.items_default_values);
        }
        if (data.hasOwnProperty('countries_list')) {
          obj.countries_list = _ApiClient["default"].convertToType(data.countries_list, ['String']);
        }
        if (data.hasOwnProperty('currencies_list')) {
          obj.currencies_list = _ApiClient["default"].convertToType(data.currencies_list, [_Currency["default"]]);
        }
        if (data.hasOwnProperty('templates_list')) {
          obj.templates_list = _ApiClient["default"].convertToType(data.templates_list, [_DocumentTemplate["default"]]);
        }
        if (data.hasOwnProperty('dn_templates_list')) {
          obj.dn_templates_list = _ApiClient["default"].convertToType(data.dn_templates_list, [_DocumentTemplate["default"]]);
        }
        if (data.hasOwnProperty('ai_templates_list')) {
          obj.ai_templates_list = _ApiClient["default"].convertToType(data.ai_templates_list, [_DocumentTemplate["default"]]);
        }
        if (data.hasOwnProperty('payment_methods_list')) {
          obj.payment_methods_list = _ApiClient["default"].convertToType(data.payment_methods_list, [_PaymentMethod["default"]]);
        }
        if (data.hasOwnProperty('payment_accounts_list')) {
          obj.payment_accounts_list = _ApiClient["default"].convertToType(data.payment_accounts_list, [_PaymentAccount["default"]]);
        }
        if (data.hasOwnProperty('vat_types_list')) {
          obj.vat_types_list = _ApiClient["default"].convertToType(data.vat_types_list, [_VatType["default"]]);
        }
        if (data.hasOwnProperty('languages_list')) {
          obj.languages_list = _ApiClient["default"].convertToType(data.languages_list, [_Language["default"]]);
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>IssuedDocumentPreCreateInfo</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>IssuedDocumentPreCreateInfo</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `default_values`
      if (data.default_values) {
        // data not null
        _IssuedDocumentPreCreateInfoDefaultValues["default"].validateJSON(data.default_values);
      }
      // validate the optional field `extra_data_default_values`
      if (data.extra_data_default_values) {
        // data not null
        _IssuedDocumentPreCreateInfoExtraDataDefaultValues["default"].validateJSON(data.extra_data_default_values);
      }
      // validate the optional field `items_default_values`
      if (data.items_default_values) {
        // data not null
        _IssuedDocumentPreCreateInfoItemsDefaultValues["default"].validateJSON(data.items_default_values);
      }
      // ensure the json data is an array
      if (!Array.isArray(data.countries_list)) {
        throw new Error('Expected the field `countries_list` to be an array in the JSON data but got ' + data.countries_list);
      }
      if (data.currencies_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.currencies_list)) {
          throw new Error('Expected the field `currencies_list` to be an array in the JSON data but got ' + data.currencies_list);
        }
        // validate the optional field `currencies_list` (array)
        var _iterator = _createForOfIteratorHelper(data.currencies_list),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _Currency["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
      }
      if (data.templates_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.templates_list)) {
          throw new Error('Expected the field `templates_list` to be an array in the JSON data but got ' + data.templates_list);
        }
        // validate the optional field `templates_list` (array)
        var _iterator2 = _createForOfIteratorHelper(data.templates_list),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;
            _DocumentTemplate["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
      }
      if (data.dn_templates_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.dn_templates_list)) {
          throw new Error('Expected the field `dn_templates_list` to be an array in the JSON data but got ' + data.dn_templates_list);
        }
        // validate the optional field `dn_templates_list` (array)
        var _iterator3 = _createForOfIteratorHelper(data.dn_templates_list),
          _step3;
        try {
          for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
            var _item2 = _step3.value;
            _DocumentTemplate["default"].validateJSON(_item2);
          }
        } catch (err) {
          _iterator3.e(err);
        } finally {
          _iterator3.f();
        }
      }
      if (data.ai_templates_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.ai_templates_list)) {
          throw new Error('Expected the field `ai_templates_list` to be an array in the JSON data but got ' + data.ai_templates_list);
        }
        // validate the optional field `ai_templates_list` (array)
        var _iterator4 = _createForOfIteratorHelper(data.ai_templates_list),
          _step4;
        try {
          for (_iterator4.s(); !(_step4 = _iterator4.n()).done;) {
            var _item3 = _step4.value;
            _DocumentTemplate["default"].validateJSON(_item3);
          }
        } catch (err) {
          _iterator4.e(err);
        } finally {
          _iterator4.f();
        }
      }
      if (data.payment_methods_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.payment_methods_list)) {
          throw new Error('Expected the field `payment_methods_list` to be an array in the JSON data but got ' + data.payment_methods_list);
        }
        // validate the optional field `payment_methods_list` (array)
        var _iterator5 = _createForOfIteratorHelper(data.payment_methods_list),
          _step5;
        try {
          for (_iterator5.s(); !(_step5 = _iterator5.n()).done;) {
            var _item4 = _step5.value;
            _PaymentMethod["default"].validateJSON(_item4);
          }
        } catch (err) {
          _iterator5.e(err);
        } finally {
          _iterator5.f();
        }
      }
      if (data.payment_accounts_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.payment_accounts_list)) {
          throw new Error('Expected the field `payment_accounts_list` to be an array in the JSON data but got ' + data.payment_accounts_list);
        }
        // validate the optional field `payment_accounts_list` (array)
        var _iterator6 = _createForOfIteratorHelper(data.payment_accounts_list),
          _step6;
        try {
          for (_iterator6.s(); !(_step6 = _iterator6.n()).done;) {
            var _item5 = _step6.value;
            _PaymentAccount["default"].validateJSON(_item5);
          }
        } catch (err) {
          _iterator6.e(err);
        } finally {
          _iterator6.f();
        }
      }
      if (data.vat_types_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.vat_types_list)) {
          throw new Error('Expected the field `vat_types_list` to be an array in the JSON data but got ' + data.vat_types_list);
        }
        // validate the optional field `vat_types_list` (array)
        var _iterator7 = _createForOfIteratorHelper(data.vat_types_list),
          _step7;
        try {
          for (_iterator7.s(); !(_step7 = _iterator7.n()).done;) {
            var _item6 = _step7.value;
            _VatType["default"].validateJSON(_item6);
          }
        } catch (err) {
          _iterator7.e(err);
        } finally {
          _iterator7.f();
        }
      }
      if (data.languages_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.languages_list)) {
          throw new Error('Expected the field `languages_list` to be an array in the JSON data but got ' + data.languages_list);
        }
        // validate the optional field `languages_list` (array)
        var _iterator8 = _createForOfIteratorHelper(data.languages_list),
          _step8;
        try {
          for (_iterator8.s(); !(_step8 = _iterator8.n()).done;) {
            var _item7 = _step8.value;
            _Language["default"].validateJSON(_item7);
          }
        } catch (err) {
          _iterator8.e(err);
        } finally {
          _iterator8.f();
        }
      }
      return true;
    }
  }]);
  return IssuedDocumentPreCreateInfo;
}();
/**
 * @member {Object.<String, Object.<String, Number>>} numerations
 */
IssuedDocumentPreCreateInfo.prototype.numerations = undefined;

/**
 * @member {Object.<String, Object.<String, Number>>} dn_numerations
 */
IssuedDocumentPreCreateInfo.prototype.dn_numerations = undefined;

/**
 * @member {module:model/IssuedDocumentPreCreateInfoDefaultValues} default_values
 */
IssuedDocumentPreCreateInfo.prototype.default_values = undefined;

/**
 * @member {module:model/IssuedDocumentPreCreateInfoExtraDataDefaultValues} extra_data_default_values
 */
IssuedDocumentPreCreateInfo.prototype.extra_data_default_values = undefined;

/**
 * @member {module:model/IssuedDocumentPreCreateInfoItemsDefaultValues} items_default_values
 */
IssuedDocumentPreCreateInfo.prototype.items_default_values = undefined;

/**
 * Countries list
 * @member {Array.<String>} countries_list
 */
IssuedDocumentPreCreateInfo.prototype.countries_list = undefined;

/**
 * Currencies list
 * @member {Array.<module:model/Currency>} currencies_list
 */
IssuedDocumentPreCreateInfo.prototype.currencies_list = undefined;

/**
 * Document templates list
 * @member {Array.<module:model/DocumentTemplate>} templates_list
 */
IssuedDocumentPreCreateInfo.prototype.templates_list = undefined;

/**
 * Delivery note templates list
 * @member {Array.<module:model/DocumentTemplate>} dn_templates_list
 */
IssuedDocumentPreCreateInfo.prototype.dn_templates_list = undefined;

/**
 * Accompanying invoice templates list
 * @member {Array.<module:model/DocumentTemplate>} ai_templates_list
 */
IssuedDocumentPreCreateInfo.prototype.ai_templates_list = undefined;

/**
 * Payment methods list
 * @member {Array.<module:model/PaymentMethod>} payment_methods_list
 */
IssuedDocumentPreCreateInfo.prototype.payment_methods_list = undefined;

/**
 * Payment accounts list
 * @member {Array.<module:model/PaymentAccount>} payment_accounts_list
 */
IssuedDocumentPreCreateInfo.prototype.payment_accounts_list = undefined;

/**
 * Vat types list
 * @member {Array.<module:model/VatType>} vat_types_list
 */
IssuedDocumentPreCreateInfo.prototype.vat_types_list = undefined;

/**
 * Languages list
 * @member {Array.<module:model/Language>} languages_list
 */
IssuedDocumentPreCreateInfo.prototype.languages_list = undefined;
var _default = exports["default"] = IssuedDocumentPreCreateInfo;
