// @fattureincloud/fattureincloud-js-sdk/dist/model/ListEmailsResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Email = _interopRequireDefault(require("./Email"));
var _ListEmailsResponsePage = _interopRequireDefault(require("./ListEmailsResponsePage"));
var _Pagination = _interopRequireDefault(require("./Pagination"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ListEmailsResponse model module.
 * @module model/ListEmailsResponse
 * @version 2.0.25
 */
var ListEmailsResponse = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>ListEmailsResponse</code>.
     * @alias module:model/ListEmailsResponse
     * @implements module:model/Pagination
     * @implements module:model/ListEmailsResponsePage
     */
  function ListEmailsResponse() {
    _classCallCheck(this, ListEmailsResponse);
    _Pagination["default"].initialize(this);
    _ListEmailsResponsePage["default"].initialize(this);
    ListEmailsResponse.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(ListEmailsResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>ListEmailsResponse</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/ListEmailsResponse} obj Optional instance to populate.
       * @return {module:model/ListEmailsResponse} The populated <code>ListEmailsResponse</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListEmailsResponse();
        _Pagination["default"].constructFromObject(data, obj);
        _ListEmailsResponsePage["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('current_page')) {
          obj.current_page = _ApiClient["default"].convertToType(data.current_page, 'Number');
        }
        if (data.hasOwnProperty('first_page_url')) {
          obj.first_page_url = _ApiClient["default"].convertToType(data.first_page_url, 'String');
        }
        if (data.hasOwnProperty('from')) {
          obj.from = _ApiClient["default"].convertToType(data.from, 'Number');
        }
        if (data.hasOwnProperty('last_page')) {
          obj.last_page = _ApiClient["default"].convertToType(data.last_page, 'Number');
        }
        if (data.hasOwnProperty('last_page_url')) {
          obj.last_page_url = _ApiClient["default"].convertToType(data.last_page_url, 'String');
        }
        if (data.hasOwnProperty('next_page_url')) {
          obj.next_page_url = _ApiClient["default"].convertToType(data.next_page_url, 'String');
        }
        if (data.hasOwnProperty('path')) {
          obj.path = _ApiClient["default"].convertToType(data.path, 'String');
        }
        if (data.hasOwnProperty('per_page')) {
          obj.per_page = _ApiClient["default"].convertToType(data.per_page, 'Number');
        }
        if (data.hasOwnProperty('prev_page_url')) {
          obj.prev_page_url = _ApiClient["default"].convertToType(data.prev_page_url, 'String');
        }
        if (data.hasOwnProperty('to')) {
          obj.to = _ApiClient["default"].convertToType(data.to, 'Number');
        }
        if (data.hasOwnProperty('total')) {
          obj.total = _ApiClient["default"].convertToType(data.total, 'Number');
        }
        if (data.hasOwnProperty('data')) {
          obj.data = _ApiClient["default"].convertToType(data.data, [_Email["default"]]);
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>ListEmailsResponse</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ListEmailsResponse</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.first_page_url && !(typeof data.first_page_url === 'string' || data.first_page_url instanceof String)) {
        throw new Error('Expected the field `first_page_url` to be a primitive type in the JSON string but got ' + data.first_page_url);
      }
      // ensure the json data is a string
      if (data.last_page_url && !(typeof data.last_page_url === 'string' || data.last_page_url instanceof String)) {
        throw new Error('Expected the field `last_page_url` to be a primitive type in the JSON string but got ' + data.last_page_url);
      }
      // ensure the json data is a string
      if (data.next_page_url && !(typeof data.next_page_url === 'string' || data.next_page_url instanceof String)) {
        throw new Error('Expected the field `next_page_url` to be a primitive type in the JSON string but got ' + data.next_page_url);
      }
      // ensure the json data is a string
      if (data.path && !(typeof data.path === 'string' || data.path instanceof String)) {
        throw new Error('Expected the field `path` to be a primitive type in the JSON string but got ' + data.path);
      }
      // ensure the json data is a string
      if (data.prev_page_url && !(typeof data.prev_page_url === 'string' || data.prev_page_url instanceof String)) {
        throw new Error('Expected the field `prev_page_url` to be a primitive type in the JSON string but got ' + data.prev_page_url);
      }
      if (data.data) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.data)) {
          throw new Error('Expected the field `data` to be an array in the JSON data but got ' + data.data);
        }
        // validate the optional field `data` (array)
        var _iterator = _createForOfIteratorHelper(data.data),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _Email["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
      }
      return true;
    }
  }]);
  return ListEmailsResponse;
}();
/**
 * Current page number.
 * @member {Number} current_page
 */
ListEmailsResponse.prototype.current_page = undefined;

/**
 * First page url.
 * @member {String} first_page_url
 */
ListEmailsResponse.prototype.first_page_url = undefined;

/**
 * First result of the page.
 * @member {Number} from
 */
ListEmailsResponse.prototype.from = undefined;

/**
 * Last page number.
 * @member {Number} last_page
 */
ListEmailsResponse.prototype.last_page = undefined;

/**
 * Last page url.
 * @member {String} last_page_url
 */
ListEmailsResponse.prototype.last_page_url = undefined;

/**
 * Next page url
 * @member {String} next_page_url
 */
ListEmailsResponse.prototype.next_page_url = undefined;

/**
 * Request path.
 * @member {String} path
 */
ListEmailsResponse.prototype.path = undefined;

/**
 * Number of result per page.
 * @member {Number} per_page
 */
ListEmailsResponse.prototype.per_page = undefined;

/**
 * Previous page url.
 * @member {String} prev_page_url
 */
ListEmailsResponse.prototype.prev_page_url = undefined;

/**
 * Last result of the page.
 * @member {Number} to
 */
ListEmailsResponse.prototype.to = undefined;

/**
 * Total number of results
 * @member {Number} total
 */
ListEmailsResponse.prototype.total = undefined;

/**
 * @member {Array.<module:model/Email>} data
 */
ListEmailsResponse.prototype.data = undefined;

// Implement Pagination interface:
/**
 * Current page number.
 * @member {Number} current_page
 */
_Pagination["default"].prototype.current_page = undefined;
/**
 * First page url.
 * @member {String} first_page_url
 */
_Pagination["default"].prototype.first_page_url = undefined;
/**
 * First result of the page.
 * @member {Number} from
 */
_Pagination["default"].prototype.from = undefined;
/**
 * Last page number.
 * @member {Number} last_page
 */
_Pagination["default"].prototype.last_page = undefined;
/**
 * Last page url.
 * @member {String} last_page_url
 */
_Pagination["default"].prototype.last_page_url = undefined;
/**
 * Next page url
 * @member {String} next_page_url
 */
_Pagination["default"].prototype.next_page_url = undefined;
/**
 * Request path.
 * @member {String} path
 */
_Pagination["default"].prototype.path = undefined;
/**
 * Number of result per page.
 * @member {Number} per_page
 */
_Pagination["default"].prototype.per_page = undefined;
/**
 * Previous page url.
 * @member {String} prev_page_url
 */
_Pagination["default"].prototype.prev_page_url = undefined;
/**
 * Last result of the page.
 * @member {Number} to
 */
_Pagination["default"].prototype.to = undefined;
/**
 * Total number of results
 * @member {Number} total
 */
_Pagination["default"].prototype.total = undefined;
// Implement ListEmailsResponsePage interface:
/**
 * @member {Array.<module:model/Email>} data
 */
_ListEmailsResponsePage["default"].prototype.data = undefined;
var _default = exports["default"] = ListEmailsResponse;
