// @fattureincloud/fattureincloud-js-sdk/dist/model/ListF24ResponseAggregation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ListF24ResponseAggregatedData = _interopRequireDefault(require("./ListF24ResponseAggregatedData"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ListF24ResponseAggregation model module.
 * @module model/ListF24ResponseAggregation
 * @version 2.0.25
 */
var ListF24ResponseAggregation = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>ListF24ResponseAggregation</code>.
     * @alias module:model/ListF24ResponseAggregation
     */
  function ListF24ResponseAggregation() {
    _classCallCheck(this, ListF24ResponseAggregation);
    ListF24ResponseAggregation.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(ListF24ResponseAggregation, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>ListF24ResponseAggregation</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/ListF24ResponseAggregation} obj Optional instance to populate.
       * @return {module:model/ListF24ResponseAggregation} The populated <code>ListF24ResponseAggregation</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListF24ResponseAggregation();
        if (data.hasOwnProperty('aggregated_data')) {
          obj.aggregated_data = _ListF24ResponseAggregatedData["default"].constructFromObject(data.aggregated_data);
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>ListF24ResponseAggregation</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ListF24ResponseAggregation</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `aggregated_data`
      if (data.aggregated_data) {
        // data not null
        _ListF24ResponseAggregatedData["default"].validateJSON(data.aggregated_data);
      }
      return true;
    }
  }]);
  return ListF24ResponseAggregation;
}();
/**
 * @member {module:model/ListF24ResponseAggregatedData} aggregated_data
 */
ListF24ResponseAggregation.prototype.aggregated_data = undefined;
var _default = exports["default"] = ListF24ResponseAggregation;
