// @fattureincloud/fattureincloud-js-sdk/dist/model/Pagination.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The Pagination model module.
 * @module model/Pagination
 * @version 2.0.25
 */
var Pagination = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>Pagination</code>.
     *
     * @alias module:model/Pagination
     */
  function Pagination() {
    _classCallCheck(this, Pagination);
    Pagination.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(Pagination, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>Pagination</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/Pagination} obj Optional instance to populate.
       * @return {module:model/Pagination} The populated <code>Pagination</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Pagination();
        if (data.hasOwnProperty('current_page')) {
          obj.current_page = _ApiClient["default"].convertToType(data.current_page, 'Number');
        }
        if (data.hasOwnProperty('first_page_url')) {
          obj.first_page_url = _ApiClient["default"].convertToType(data.first_page_url, 'String');
        }
        if (data.hasOwnProperty('from')) {
          obj.from = _ApiClient["default"].convertToType(data.from, 'Number');
        }
        if (data.hasOwnProperty('last_page')) {
          obj.last_page = _ApiClient["default"].convertToType(data.last_page, 'Number');
        }
        if (data.hasOwnProperty('last_page_url')) {
          obj.last_page_url = _ApiClient["default"].convertToType(data.last_page_url, 'String');
        }
        if (data.hasOwnProperty('next_page_url')) {
          obj.next_page_url = _ApiClient["default"].convertToType(data.next_page_url, 'String');
        }
        if (data.hasOwnProperty('path')) {
          obj.path = _ApiClient["default"].convertToType(data.path, 'String');
        }
        if (data.hasOwnProperty('per_page')) {
          obj.per_page = _ApiClient["default"].convertToType(data.per_page, 'Number');
        }
        if (data.hasOwnProperty('prev_page_url')) {
          obj.prev_page_url = _ApiClient["default"].convertToType(data.prev_page_url, 'String');
        }
        if (data.hasOwnProperty('to')) {
          obj.to = _ApiClient["default"].convertToType(data.to, 'Number');
        }
        if (data.hasOwnProperty('total')) {
          obj.total = _ApiClient["default"].convertToType(data.total, 'Number');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>Pagination</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Pagination</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.first_page_url && !(typeof data.first_page_url === 'string' || data.first_page_url instanceof String)) {
        throw new Error('Expected the field `first_page_url` to be a primitive type in the JSON string but got ' + data.first_page_url);
      }
      // ensure the json data is a string
      if (data.last_page_url && !(typeof data.last_page_url === 'string' || data.last_page_url instanceof String)) {
        throw new Error('Expected the field `last_page_url` to be a primitive type in the JSON string but got ' + data.last_page_url);
      }
      // ensure the json data is a string
      if (data.next_page_url && !(typeof data.next_page_url === 'string' || data.next_page_url instanceof String)) {
        throw new Error('Expected the field `next_page_url` to be a primitive type in the JSON string but got ' + data.next_page_url);
      }
      // ensure the json data is a string
      if (data.path && !(typeof data.path === 'string' || data.path instanceof String)) {
        throw new Error('Expected the field `path` to be a primitive type in the JSON string but got ' + data.path);
      }
      // ensure the json data is a string
      if (data.prev_page_url && !(typeof data.prev_page_url === 'string' || data.prev_page_url instanceof String)) {
        throw new Error('Expected the field `prev_page_url` to be a primitive type in the JSON string but got ' + data.prev_page_url);
      }
      return true;
    }
  }]);
  return Pagination;
}();
/**
 * Current page number.
 * @member {Number} current_page
 */
Pagination.prototype.current_page = undefined;

/**
 * First page url.
 * @member {String} first_page_url
 */
Pagination.prototype.first_page_url = undefined;

/**
 * First result of the page.
 * @member {Number} from
 */
Pagination.prototype.from = undefined;

/**
 * Last page number.
 * @member {Number} last_page
 */
Pagination.prototype.last_page = undefined;

/**
 * Last page url.
 * @member {String} last_page_url
 */
Pagination.prototype.last_page_url = undefined;

/**
 * Next page url
 * @member {String} next_page_url
 */
Pagination.prototype.next_page_url = undefined;

/**
 * Request path.
 * @member {String} path
 */
Pagination.prototype.path = undefined;

/**
 * Number of result per page.
 * @member {Number} per_page
 */
Pagination.prototype.per_page = undefined;

/**
 * Previous page url.
 * @member {String} prev_page_url
 */
Pagination.prototype.prev_page_url = undefined;

/**
 * Last result of the page.
 * @member {Number} to
 */
Pagination.prototype.to = undefined;

/**
 * Total number of results
 * @member {Number} total
 */
Pagination.prototype.total = undefined;
var _default = exports["default"] = Pagination;
