// @fattureincloud/fattureincloud-js-sdk/dist/model/PaymentAccount.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PaymentAccountType = _interopRequireDefault(require("./PaymentAccountType"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The PaymentAccount model module.
 * @module model/PaymentAccount
 * @version 2.0.25
 */
var PaymentAccount = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>PaymentAccount</code>.
     * @alias module:model/PaymentAccount
     */
  function PaymentAccount() {
    _classCallCheck(this, PaymentAccount);
    PaymentAccount.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(PaymentAccount, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>PaymentAccount</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/PaymentAccount} obj Optional instance to populate.
       * @return {module:model/PaymentAccount} The populated <code>PaymentAccount</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PaymentAccount();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'Number');
        }
        if (data.hasOwnProperty('name')) {
          obj.name = _ApiClient["default"].convertToType(data.name, 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj.type = _PaymentAccountType["default"].constructFromObject(data.type);
        }
        if (data.hasOwnProperty('iban')) {
          obj.iban = _ApiClient["default"].convertToType(data.iban, 'String');
        }
        if (data.hasOwnProperty('sia')) {
          obj.sia = _ApiClient["default"].convertToType(data.sia, 'String');
        }
        if (data.hasOwnProperty('cuc')) {
          obj.cuc = _ApiClient["default"].convertToType(data.cuc, 'String');
        }
        if (data.hasOwnProperty('virtual')) {
          obj.virtual = _ApiClient["default"].convertToType(data.virtual, 'Boolean');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>PaymentAccount</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PaymentAccount</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.name && !(typeof data.name === 'string' || data.name instanceof String)) {
        throw new Error('Expected the field `name` to be a primitive type in the JSON string but got ' + data.name);
      }
      // ensure the json data is a string
      if (data.iban && !(typeof data.iban === 'string' || data.iban instanceof String)) {
        throw new Error('Expected the field `iban` to be a primitive type in the JSON string but got ' + data.iban);
      }
      // ensure the json data is a string
      if (data.sia && !(typeof data.sia === 'string' || data.sia instanceof String)) {
        throw new Error('Expected the field `sia` to be a primitive type in the JSON string but got ' + data.sia);
      }
      // ensure the json data is a string
      if (data.cuc && !(typeof data.cuc === 'string' || data.cuc instanceof String)) {
        throw new Error('Expected the field `cuc` to be a primitive type in the JSON string but got ' + data.cuc);
      }
      return true;
    }
  }]);
  return PaymentAccount;
}();
/**
 * Payment account id
 * @member {Number} id
 */
PaymentAccount.prototype.id = undefined;

/**
 * Payment account name
 * @member {String} name
 */
PaymentAccount.prototype.name = undefined;

/**
 * @member {module:model/PaymentAccountType} type
 */
PaymentAccount.prototype.type = undefined;

/**
 * Payment account iban
 * @member {String} iban
 */
PaymentAccount.prototype.iban = undefined;

/**
 * Payment account sia
 * @member {String} sia
 */
PaymentAccount.prototype.sia = undefined;

/**
 * Payment account cuc
 * @member {String} cuc
 */
PaymentAccount.prototype.cuc = undefined;

/**
 * Payment method is virtual
 * @member {Boolean} virtual
 */
PaymentAccount.prototype.virtual = undefined;
var _default = exports["default"] = PaymentAccount;
