// @fattureincloud/fattureincloud-js-sdk/dist/model/PaymentMethod.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PaymentAccount = _interopRequireDefault(require("./PaymentAccount"));
var _PaymentMethodDetails = _interopRequireDefault(require("./PaymentMethodDetails"));
var _PaymentMethodType = _interopRequireDefault(require("./PaymentMethodType"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The PaymentMethod model module.
 * @module model/PaymentMethod
 * @version 2.0.25
 */
var PaymentMethod = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>PaymentMethod</code>.
     * @alias module:model/PaymentMethod
     */
  function PaymentMethod() {
    _classCallCheck(this, PaymentMethod);
    PaymentMethod.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(PaymentMethod, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>PaymentMethod</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/PaymentMethod} obj Optional instance to populate.
       * @return {module:model/PaymentMethod} The populated <code>PaymentMethod</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PaymentMethod();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'Number');
        }
        if (data.hasOwnProperty('name')) {
          obj.name = _ApiClient["default"].convertToType(data.name, 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj.type = _PaymentMethodType["default"].constructFromObject(data.type);
        }
        if (data.hasOwnProperty('is_default')) {
          obj.is_default = _ApiClient["default"].convertToType(data.is_default, 'Boolean');
        }
        if (data.hasOwnProperty('default_payment_account')) {
          obj.default_payment_account = _PaymentAccount["default"].constructFromObject(data.default_payment_account);
        }
        if (data.hasOwnProperty('details')) {
          obj.details = _ApiClient["default"].convertToType(data.details, [_PaymentMethodDetails["default"]]);
        }
        if (data.hasOwnProperty('bank_iban')) {
          obj.bank_iban = _ApiClient["default"].convertToType(data.bank_iban, 'String');
        }
        if (data.hasOwnProperty('bank_name')) {
          obj.bank_name = _ApiClient["default"].convertToType(data.bank_name, 'String');
        }
        if (data.hasOwnProperty('bank_beneficiary')) {
          obj.bank_beneficiary = _ApiClient["default"].convertToType(data.bank_beneficiary, 'String');
        }
        if (data.hasOwnProperty('ei_payment_method')) {
          obj.ei_payment_method = _ApiClient["default"].convertToType(data.ei_payment_method, 'String');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>PaymentMethod</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>PaymentMethod</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.name && !(typeof data.name === 'string' || data.name instanceof String)) {
        throw new Error('Expected the field `name` to be a primitive type in the JSON string but got ' + data.name);
      }
      // validate the optional field `default_payment_account`
      if (data.default_payment_account) {
        // data not null
        _PaymentAccount["default"].validateJSON(data.default_payment_account);
      }
      if (data.details) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.details)) {
          throw new Error('Expected the field `details` to be an array in the JSON data but got ' + data.details);
        }
        // validate the optional field `details` (array)
        var _iterator = _createForOfIteratorHelper(data.details),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _PaymentMethodDetails["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
      }
      // ensure the json data is a string
      if (data.bank_iban && !(typeof data.bank_iban === 'string' || data.bank_iban instanceof String)) {
        throw new Error('Expected the field `bank_iban` to be a primitive type in the JSON string but got ' + data.bank_iban);
      }
      // ensure the json data is a string
      if (data.bank_name && !(typeof data.bank_name === 'string' || data.bank_name instanceof String)) {
        throw new Error('Expected the field `bank_name` to be a primitive type in the JSON string but got ' + data.bank_name);
      }
      // ensure the json data is a string
      if (data.bank_beneficiary && !(typeof data.bank_beneficiary === 'string' || data.bank_beneficiary instanceof String)) {
        throw new Error('Expected the field `bank_beneficiary` to be a primitive type in the JSON string but got ' + data.bank_beneficiary);
      }
      // ensure the json data is a string
      if (data.ei_payment_method && !(typeof data.ei_payment_method === 'string' || data.ei_payment_method instanceof String)) {
        throw new Error('Expected the field `ei_payment_method` to be a primitive type in the JSON string but got ' + data.ei_payment_method);
      }
      return true;
    }
  }]);
  return PaymentMethod;
}();
/**
 * Payment method id
 * @member {Number} id
 */
PaymentMethod.prototype.id = undefined;

/**
 * Payment method name
 * @member {String} name
 */
PaymentMethod.prototype.name = undefined;

/**
 * @member {module:model/PaymentMethodType} type
 */
PaymentMethod.prototype.type = undefined;

/**
 * Payment method is default
 * @member {Boolean} is_default
 */
PaymentMethod.prototype.is_default = undefined;

/**
 * @member {module:model/PaymentAccount} default_payment_account
 */
PaymentMethod.prototype.default_payment_account = undefined;

/**
 * Payment method details
 * @member {Array.<module:model/PaymentMethodDetails>} details
 */
PaymentMethod.prototype.details = undefined;

/**
 * Payment method bank iban
 * @member {String} bank_iban
 */
PaymentMethod.prototype.bank_iban = undefined;

/**
 * Payment method bank name
 * @member {String} bank_name
 */
PaymentMethod.prototype.bank_name = undefined;

/**
 * Payment method bank beneficiary
 * @member {String} bank_beneficiary
 */
PaymentMethod.prototype.bank_beneficiary = undefined;

/**
 * E-invoice payment method
 * @member {String} ei_payment_method
 */
PaymentMethod.prototype.ei_payment_method = undefined;
var _default = exports["default"] = PaymentMethod;
