// @fattureincloud/fattureincloud-js-sdk/dist/model/Receipt.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PaymentAccount = _interopRequireDefault(require("./PaymentAccount"));
var _ReceiptItemsListItem = _interopRequireDefault(require("./ReceiptItemsListItem"));
var _ReceiptType = _interopRequireDefault(require("./ReceiptType"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The Receipt model module.
 * @module model/Receipt
 * @version 2.0.25
 */
var Receipt = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>Receipt</code>.
     * @alias module:model/Receipt
     */
  function Receipt() {
    _classCallCheck(this, Receipt);
    Receipt.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(Receipt, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>Receipt</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/Receipt} obj Optional instance to populate.
       * @return {module:model/Receipt} The populated <code>Receipt</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Receipt();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'Number');
        }
        if (data.hasOwnProperty('date')) {
          obj.date = _ApiClient["default"].convertToType(data.date, 'Date');
        }
        if (data.hasOwnProperty('number')) {
          obj.number = _ApiClient["default"].convertToType(data.number, 'Number');
        }
        if (data.hasOwnProperty('numeration')) {
          obj.numeration = _ApiClient["default"].convertToType(data.numeration, 'String');
        }
        if (data.hasOwnProperty('amount_net')) {
          obj.amount_net = _ApiClient["default"].convertToType(data.amount_net, 'Number');
        }
        if (data.hasOwnProperty('amount_vat')) {
          obj.amount_vat = _ApiClient["default"].convertToType(data.amount_vat, 'Number');
        }
        if (data.hasOwnProperty('amount_gross')) {
          obj.amount_gross = _ApiClient["default"].convertToType(data.amount_gross, 'Number');
        }
        if (data.hasOwnProperty('use_gross_prices')) {
          obj.use_gross_prices = _ApiClient["default"].convertToType(data.use_gross_prices, 'Boolean');
        }
        if (data.hasOwnProperty('type')) {
          obj.type = _ReceiptType["default"].constructFromObject(data.type);
        }
        if (data.hasOwnProperty('description')) {
          obj.description = _ApiClient["default"].convertToType(data.description, 'String');
        }
        if (data.hasOwnProperty('rc_center')) {
          obj.rc_center = _ApiClient["default"].convertToType(data.rc_center, 'String');
        }
        if (data.hasOwnProperty('created_at')) {
          obj.created_at = _ApiClient["default"].convertToType(data.created_at, 'String');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj.updated_at = _ApiClient["default"].convertToType(data.updated_at, 'String');
        }
        if (data.hasOwnProperty('payment_account')) {
          obj.payment_account = _PaymentAccount["default"].constructFromObject(data.payment_account);
        }
        if (data.hasOwnProperty('items_list')) {
          obj.items_list = _ApiClient["default"].convertToType(data.items_list, [_ReceiptItemsListItem["default"]]);
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>Receipt</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Receipt</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.numeration && !(typeof data.numeration === 'string' || data.numeration instanceof String)) {
        throw new Error('Expected the field `numeration` to be a primitive type in the JSON string but got ' + data.numeration);
      }
      // ensure the json data is a string
      if (data.description && !(typeof data.description === 'string' || data.description instanceof String)) {
        throw new Error('Expected the field `description` to be a primitive type in the JSON string but got ' + data.description);
      }
      // ensure the json data is a string
      if (data.rc_center && !(typeof data.rc_center === 'string' || data.rc_center instanceof String)) {
        throw new Error('Expected the field `rc_center` to be a primitive type in the JSON string but got ' + data.rc_center);
      }
      // ensure the json data is a string
      if (data.created_at && !(typeof data.created_at === 'string' || data.created_at instanceof String)) {
        throw new Error('Expected the field `created_at` to be a primitive type in the JSON string but got ' + data.created_at);
      }
      // ensure the json data is a string
      if (data.updated_at && !(typeof data.updated_at === 'string' || data.updated_at instanceof String)) {
        throw new Error('Expected the field `updated_at` to be a primitive type in the JSON string but got ' + data.updated_at);
      }
      // validate the optional field `payment_account`
      if (data.payment_account) {
        // data not null
        _PaymentAccount["default"].validateJSON(data.payment_account);
      }
      if (data.items_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.items_list)) {
          throw new Error('Expected the field `items_list` to be an array in the JSON data but got ' + data.items_list);
        }
        // validate the optional field `items_list` (array)
        var _iterator = _createForOfIteratorHelper(data.items_list),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _ReceiptItemsListItem["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
      }
      return true;
    }
  }]);
  return Receipt;
}();
/**
 * Receipt id
 * @member {Number} id
 */
Receipt.prototype.id = undefined;

/**
 * Receipt date
 * @member {Date} date
 */
Receipt.prototype.date = undefined;

/**
 * Receipt number
 * @member {Number} number
 */
Receipt.prototype.number = undefined;

/**
 * Receipt numeration
 * @member {String} numeration
 */
Receipt.prototype.numeration = undefined;

/**
 * Receipt total net amount
 * @member {Number} amount_net
 */
Receipt.prototype.amount_net = undefined;

/**
 * Receipt total vat amount
 * @member {Number} amount_vat
 */
Receipt.prototype.amount_vat = undefined;

/**
 * Receipt total gross amount
 * @member {Number} amount_gross
 */
Receipt.prototype.amount_gross = undefined;

/**
 * Receipt uses gross prices
 * @member {Boolean} use_gross_prices
 */
Receipt.prototype.use_gross_prices = undefined;

/**
 * @member {module:model/ReceiptType} type
 */
Receipt.prototype.type = undefined;

/**
 * Receipt description
 * @member {String} description
 */
Receipt.prototype.description = undefined;

/**
 * Receipt revenue center
 * @member {String} rc_center
 */
Receipt.prototype.rc_center = undefined;

/**
 * Receipt creation date
 * @member {String} created_at
 */
Receipt.prototype.created_at = undefined;

/**
 * Receipt last update date
 * @member {String} updated_at
 */
Receipt.prototype.updated_at = undefined;

/**
 * @member {module:model/PaymentAccount} payment_account
 */
Receipt.prototype.payment_account = undefined;

/**
 * @member {Array.<module:model/ReceiptItemsListItem>} items_list
 */
Receipt.prototype.items_list = undefined;
var _default = exports["default"] = Receipt;
