// @fattureincloud/fattureincloud-js-sdk/dist/model/ReceivedDocumentInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Currency = _interopRequireDefault(require("./Currency"));
var _PaymentAccount = _interopRequireDefault(require("./PaymentAccount"));
var _ReceivedDocumentInfoDefaultValues = _interopRequireDefault(require("./ReceivedDocumentInfoDefaultValues"));
var _ReceivedDocumentInfoItemsDefaultValues = _interopRequireDefault(require("./ReceivedDocumentInfoItemsDefaultValues"));
var _VatType = _interopRequireDefault(require("./VatType"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ReceivedDocumentInfo model module.
 * @module model/ReceivedDocumentInfo
 * @version 2.0.25
 */
var ReceivedDocumentInfo = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>ReceivedDocumentInfo</code>.
     * @alias module:model/ReceivedDocumentInfo
     */
  function ReceivedDocumentInfo() {
    _classCallCheck(this, ReceivedDocumentInfo);
    ReceivedDocumentInfo.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(ReceivedDocumentInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>ReceivedDocumentInfo</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/ReceivedDocumentInfo} obj Optional instance to populate.
       * @return {module:model/ReceivedDocumentInfo} The populated <code>ReceivedDocumentInfo</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ReceivedDocumentInfo();
        if (data.hasOwnProperty('default_values')) {
          obj.default_values = _ReceivedDocumentInfoDefaultValues["default"].constructFromObject(data.default_values);
        }
        if (data.hasOwnProperty('items_default_values')) {
          obj.items_default_values = _ReceivedDocumentInfoItemsDefaultValues["default"].constructFromObject(data.items_default_values);
        }
        if (data.hasOwnProperty('countries_list')) {
          obj.countries_list = _ApiClient["default"].convertToType(data.countries_list, ['String']);
        }
        if (data.hasOwnProperty('currencies_list')) {
          obj.currencies_list = _ApiClient["default"].convertToType(data.currencies_list, [_Currency["default"]]);
        }
        if (data.hasOwnProperty('categories_list')) {
          obj.categories_list = _ApiClient["default"].convertToType(data.categories_list, ['String']);
        }
        if (data.hasOwnProperty('payment_accounts_list')) {
          obj.payment_accounts_list = _ApiClient["default"].convertToType(data.payment_accounts_list, [_PaymentAccount["default"]]);
        }
        if (data.hasOwnProperty('vat_types_list')) {
          obj.vat_types_list = _ApiClient["default"].convertToType(data.vat_types_list, [_VatType["default"]]);
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>ReceivedDocumentInfo</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ReceivedDocumentInfo</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `default_values`
      if (data.default_values) {
        // data not null
        _ReceivedDocumentInfoDefaultValues["default"].validateJSON(data.default_values);
      }
      // validate the optional field `items_default_values`
      if (data.items_default_values) {
        // data not null
        _ReceivedDocumentInfoItemsDefaultValues["default"].validateJSON(data.items_default_values);
      }
      // ensure the json data is an array
      if (!Array.isArray(data.countries_list)) {
        throw new Error('Expected the field `countries_list` to be an array in the JSON data but got ' + data.countries_list);
      }
      if (data.currencies_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.currencies_list)) {
          throw new Error('Expected the field `currencies_list` to be an array in the JSON data but got ' + data.currencies_list);
        }
        // validate the optional field `currencies_list` (array)
        var _iterator = _createForOfIteratorHelper(data.currencies_list),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _Currency["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
      }
      // ensure the json data is an array
      if (!Array.isArray(data.categories_list)) {
        throw new Error('Expected the field `categories_list` to be an array in the JSON data but got ' + data.categories_list);
      }
      if (data.payment_accounts_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.payment_accounts_list)) {
          throw new Error('Expected the field `payment_accounts_list` to be an array in the JSON data but got ' + data.payment_accounts_list);
        }
        // validate the optional field `payment_accounts_list` (array)
        var _iterator2 = _createForOfIteratorHelper(data.payment_accounts_list),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;
            _PaymentAccount["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
      }
      if (data.vat_types_list) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data.vat_types_list)) {
          throw new Error('Expected the field `vat_types_list` to be an array in the JSON data but got ' + data.vat_types_list);
        }
        // validate the optional field `vat_types_list` (array)
        var _iterator3 = _createForOfIteratorHelper(data.vat_types_list),
          _step3;
        try {
          for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
            var _item2 = _step3.value;
            _VatType["default"].validateJSON(_item2);
          }
        } catch (err) {
          _iterator3.e(err);
        } finally {
          _iterator3.f();
        }
      }
      return true;
    }
  }]);
  return ReceivedDocumentInfo;
}();
/**
 * @member {module:model/ReceivedDocumentInfoDefaultValues} default_values
 */
ReceivedDocumentInfo.prototype.default_values = undefined;

/**
 * @member {module:model/ReceivedDocumentInfoItemsDefaultValues} items_default_values
 */
ReceivedDocumentInfo.prototype.items_default_values = undefined;

/**
 * Countries list
 * @member {Array.<String>} countries_list
 */
ReceivedDocumentInfo.prototype.countries_list = undefined;

/**
 * Currencies list
 * @member {Array.<module:model/Currency>} currencies_list
 */
ReceivedDocumentInfo.prototype.currencies_list = undefined;

/**
 * Categories list
 * @member {Array.<String>} categories_list
 */
ReceivedDocumentInfo.prototype.categories_list = undefined;

/**
 * Payments accounts list
 * @member {Array.<module:model/PaymentAccount>} payment_accounts_list
 */
ReceivedDocumentInfo.prototype.payment_accounts_list = undefined;

/**
 * Vat types list
 * @member {Array.<module:model/VatType>} vat_types_list
 */
ReceivedDocumentInfo.prototype.vat_types_list = undefined;
var _default = exports["default"] = ReceivedDocumentInfo;
