// @fattureincloud/fattureincloud-js-sdk/dist/model/ReceivedDocumentPaymentsListItem.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PaymentAccount = _interopRequireDefault(require("./PaymentAccount"));
var _ReceivedDocumentPaymentsListItemPaymentTerms = _interopRequireDefault(require("./ReceivedDocumentPaymentsListItemPaymentTerms"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The ReceivedDocumentPaymentsListItem model module.
 * @module model/ReceivedDocumentPaymentsListItem
 * @version 2.0.25
 */
var ReceivedDocumentPaymentsListItem = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>ReceivedDocumentPaymentsListItem</code>.
     * @alias module:model/ReceivedDocumentPaymentsListItem
     */
  function ReceivedDocumentPaymentsListItem() {
    _classCallCheck(this, ReceivedDocumentPaymentsListItem);
    ReceivedDocumentPaymentsListItem.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(ReceivedDocumentPaymentsListItem, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>ReceivedDocumentPaymentsListItem</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/ReceivedDocumentPaymentsListItem} obj Optional instance to populate.
       * @return {module:model/ReceivedDocumentPaymentsListItem} The populated <code>ReceivedDocumentPaymentsListItem</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ReceivedDocumentPaymentsListItem();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'Number');
        }
        if (data.hasOwnProperty('amount')) {
          obj.amount = _ApiClient["default"].convertToType(data.amount, 'Number');
        }
        if (data.hasOwnProperty('due_date')) {
          obj.due_date = _ApiClient["default"].convertToType(data.due_date, 'Date');
        }
        if (data.hasOwnProperty('paid_date')) {
          obj.paid_date = _ApiClient["default"].convertToType(data.paid_date, 'Date');
        }
        if (data.hasOwnProperty('payment_terms')) {
          obj.payment_terms = _ReceivedDocumentPaymentsListItemPaymentTerms["default"].constructFromObject(data.payment_terms);
        }
        if (data.hasOwnProperty('status')) {
          obj.status = _ApiClient["default"].convertToType(data.status, 'String');
        }
        if (data.hasOwnProperty('payment_account')) {
          obj.payment_account = _PaymentAccount["default"].constructFromObject(data.payment_account);
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>ReceivedDocumentPaymentsListItem</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ReceivedDocumentPaymentsListItem</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `payment_terms`
      if (data.payment_terms) {
        // data not null
        _ReceivedDocumentPaymentsListItemPaymentTerms["default"].validateJSON(data.payment_terms);
      }
      // ensure the json data is a string
      if (data.status && !(typeof data.status === 'string' || data.status instanceof String)) {
        throw new Error('Expected the field `status` to be a primitive type in the JSON string but got ' + data.status);
      }
      // validate the optional field `payment_account`
      if (data.payment_account) {
        // data not null
        _PaymentAccount["default"].validateJSON(data.payment_account);
      }
      return true;
    }
  }]);
  return ReceivedDocumentPaymentsListItem;
}();
/**
 * Received document payment id
 * @member {Number} id
 */
ReceivedDocumentPaymentsListItem.prototype.id = undefined;

/**
 * Received document payment total amount
 * @member {Number} amount
 */
ReceivedDocumentPaymentsListItem.prototype.amount = undefined;

/**
 * Due date
 * @member {Date} due_date
 */
ReceivedDocumentPaymentsListItem.prototype.due_date = undefined;

/**
 * Received document payment paid date
 * @member {Date} paid_date
 */
ReceivedDocumentPaymentsListItem.prototype.paid_date = undefined;

/**
 * @member {module:model/ReceivedDocumentPaymentsListItemPaymentTerms} payment_terms
 */
ReceivedDocumentPaymentsListItem.prototype.payment_terms = undefined;

/**
 * Received document payment status
 * @member {String} status
 */
ReceivedDocumentPaymentsListItem.prototype.status = undefined;

/**
 * @member {module:model/PaymentAccount} payment_account
 */
ReceivedDocumentPaymentsListItem.prototype.payment_account = undefined;
var _default = exports["default"] = ReceivedDocumentPaymentsListItem;
