// @fattureincloud/fattureincloud-js-sdk/dist/model/User.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The User model module.
 * @module model/User
 * @version 2.0.25
 */
var User = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>User</code>.
     * @alias module:model/User
     */
  function User() {
    _classCallCheck(this, User);
    User.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(User, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>User</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/User} obj Optional instance to populate.
       * @return {module:model/User} The populated <code>User</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new User();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'Number');
        }
        if (data.hasOwnProperty('name')) {
          obj.name = _ApiClient["default"].convertToType(data.name, 'String');
        }
        if (data.hasOwnProperty('first_name')) {
          obj.first_name = _ApiClient["default"].convertToType(data.first_name, 'String');
        }
        if (data.hasOwnProperty('last_name')) {
          obj.last_name = _ApiClient["default"].convertToType(data.last_name, 'String');
        }
        if (data.hasOwnProperty('email')) {
          obj.email = _ApiClient["default"].convertToType(data.email, 'String');
        }
        if (data.hasOwnProperty('hash')) {
          obj.hash = _ApiClient["default"].convertToType(data.hash, 'String');
        }
        if (data.hasOwnProperty('picture')) {
          obj.picture = _ApiClient["default"].convertToType(data.picture, 'String');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>User</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>User</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.name && !(typeof data.name === 'string' || data.name instanceof String)) {
        throw new Error('Expected the field `name` to be a primitive type in the JSON string but got ' + data.name);
      }
      // ensure the json data is a string
      if (data.first_name && !(typeof data.first_name === 'string' || data.first_name instanceof String)) {
        throw new Error('Expected the field `first_name` to be a primitive type in the JSON string but got ' + data.first_name);
      }
      // ensure the json data is a string
      if (data.last_name && !(typeof data.last_name === 'string' || data.last_name instanceof String)) {
        throw new Error('Expected the field `last_name` to be a primitive type in the JSON string but got ' + data.last_name);
      }
      // ensure the json data is a string
      if (data.email && !(typeof data.email === 'string' || data.email instanceof String)) {
        throw new Error('Expected the field `email` to be a primitive type in the JSON string but got ' + data.email);
      }
      // ensure the json data is a string
      if (data.hash && !(typeof data.hash === 'string' || data.hash instanceof String)) {
        throw new Error('Expected the field `hash` to be a primitive type in the JSON string but got ' + data.hash);
      }
      // ensure the json data is a string
      if (data.picture && !(typeof data.picture === 'string' || data.picture instanceof String)) {
        throw new Error('Expected the field `picture` to be a primitive type in the JSON string but got ' + data.picture);
      }
      return true;
    }
  }]);
  return User;
}();
/**
 * User id
 * @member {Number} id
 */
User.prototype.id = undefined;

/**
 * User full name
 * @member {String} name
 */
User.prototype.name = undefined;

/**
 * User first name
 * @member {String} first_name
 */
User.prototype.first_name = undefined;

/**
 * User last name
 * @member {String} last_name
 */
User.prototype.last_name = undefined;

/**
 * User email address
 * @member {String} email
 */
User.prototype.email = undefined;

/**
 * User hash
 * @member {String} hash
 */
User.prototype.hash = undefined;

/**
 * User picture
 * @member {String} picture
 */
User.prototype.picture = undefined;
var _default = exports["default"] = User;
