// @fattureincloud/fattureincloud-js-sdk/dist/model/VatType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The VatType model module.
 * @module model/VatType
 * @version 2.0.25
 */
var VatType = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>VatType</code>.
     * @alias module:model/VatType
     */
  function VatType() {
    _classCallCheck(this, VatType);
    VatType.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(VatType, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>VatType</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/VatType} obj Optional instance to populate.
       * @return {module:model/VatType} The populated <code>VatType</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new VatType();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'Number');
        }
        if (data.hasOwnProperty('value')) {
          obj.value = _ApiClient["default"].convertToType(data.value, 'Number');
        }
        if (data.hasOwnProperty('description')) {
          obj.description = _ApiClient["default"].convertToType(data.description, 'String');
        }
        if (data.hasOwnProperty('notes')) {
          obj.notes = _ApiClient["default"].convertToType(data.notes, 'String');
        }
        if (data.hasOwnProperty('e_invoice')) {
          obj.e_invoice = _ApiClient["default"].convertToType(data.e_invoice, 'Boolean');
        }
        if (data.hasOwnProperty('ei_type')) {
          obj.ei_type = _ApiClient["default"].convertToType(data.ei_type, 'String');
        }
        if (data.hasOwnProperty('ei_description')) {
          obj.ei_description = _ApiClient["default"].convertToType(data.ei_description, 'String');
        }
        if (data.hasOwnProperty('editable')) {
          obj.editable = _ApiClient["default"].convertToType(data.editable, 'Boolean');
        }
        if (data.hasOwnProperty('is_disabled')) {
          obj.is_disabled = _ApiClient["default"].convertToType(data.is_disabled, 'Boolean');
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>VatType</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>VatType</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.description && !(typeof data.description === 'string' || data.description instanceof String)) {
        throw new Error('Expected the field `description` to be a primitive type in the JSON string but got ' + data.description);
      }
      // ensure the json data is a string
      if (data.notes && !(typeof data.notes === 'string' || data.notes instanceof String)) {
        throw new Error('Expected the field `notes` to be a primitive type in the JSON string but got ' + data.notes);
      }
      // ensure the json data is a string
      if (data.ei_type && !(typeof data.ei_type === 'string' || data.ei_type instanceof String)) {
        throw new Error('Expected the field `ei_type` to be a primitive type in the JSON string but got ' + data.ei_type);
      }
      // ensure the json data is a string
      if (data.ei_description && !(typeof data.ei_description === 'string' || data.ei_description instanceof String)) {
        throw new Error('Expected the field `ei_description` to be a primitive type in the JSON string but got ' + data.ei_description);
      }
      return true;
    }
  }]);
  return VatType;
}();
/**
 * Vat type id
 * @member {Number} id
 */
VatType.prototype.id = undefined;

/**
 * [Read Only] Vat type percentual value
 * @member {Number} value
 */
VatType.prototype.value = undefined;

/**
 * Vat type short description
 * @member {String} description
 */
VatType.prototype.description = undefined;

/**
 * Vat type notes shown in documents
 * @member {String} notes
 */
VatType.prototype.notes = undefined;

/**
 * Vat type is usable for e-invoices
 * @member {Boolean} e_invoice
 */
VatType.prototype.e_invoice = undefined;

/**
 * Vat type e-invoice type (natura)
 * @member {String} ei_type
 */
VatType.prototype.ei_type = undefined;

/**
 * Vat type e-invoice description
 * @member {String} ei_description
 */
VatType.prototype.ei_description = undefined;

/**
 * [Read Only] Is the vat type is editable.
 * @member {Boolean} editable
 */
VatType.prototype.editable = undefined;

/**
 * Is the vat type disabled
 * @member {Boolean} is_disabled
 */
VatType.prototype.is_disabled = undefined;
var _default = exports["default"] = VatType;
