// @fattureincloud/fattureincloud-js-sdk/dist/model/WebhooksSubscription.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _EventType = _interopRequireDefault(require("./EventType"));
var _WebhooksSubscriptionConfig = _interopRequireDefault(require("./WebhooksSubscriptionConfig"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * Fatture in Cloud API v2 - API Reference
 * Connect your software with Fatture in Cloud, the invoicing platform chosen by more than 500.000 businesses in Italy.   The Fatture in Cloud API is based on REST, and makes possible to interact with the user related data prior authorization via OAuth2 protocol.
 *
 * The version of the OpenAPI document: 2.0.32
 * Contact: info@fattureincloud.it
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The WebhooksSubscription model module.
 * @module model/WebhooksSubscription
 * @version 2.0.25
 */
var WebhooksSubscription = /*#__PURE__*/function () {
  /**
     * Constructs a new <code>WebhooksSubscription</code>.
     * @alias module:model/WebhooksSubscription
     */
  function WebhooksSubscription() {
    _classCallCheck(this, WebhooksSubscription);
    WebhooksSubscription.initialize(this);
  }

  /**
     * Initializes the fields of this object.
     * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
     * Only for internal use.
     */
  _createClass(WebhooksSubscription, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
       * Constructs a <code>WebhooksSubscription</code> from a plain JavaScript object, optionally creating a new instance.
       * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @param {module:model/WebhooksSubscription} obj Optional instance to populate.
       * @return {module:model/WebhooksSubscription} The populated <code>WebhooksSubscription</code> instance.
       */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WebhooksSubscription();
        if (data.hasOwnProperty('id')) {
          obj.id = _ApiClient["default"].convertToType(data.id, 'String');
        }
        if (data.hasOwnProperty('sink')) {
          obj.sink = _ApiClient["default"].convertToType(data.sink, 'String');
        }
        if (data.hasOwnProperty('verified')) {
          obj.verified = _ApiClient["default"].convertToType(data.verified, 'Boolean');
        }
        if (data.hasOwnProperty('types')) {
          obj.types = _ApiClient["default"].convertToType(data.types, [_EventType["default"]]);
        }
        if (data.hasOwnProperty('config')) {
          obj.config = _WebhooksSubscriptionConfig["default"].constructFromObject(data.config);
        }
      }
      return obj;
    }

    /**
       * Validates the JSON data with respect to <code>WebhooksSubscription</code>.
       * @param {Object} data The plain JavaScript object bearing properties of interest.
       * @return {boolean} to indicate whether the JSON data is valid with respect to <code>WebhooksSubscription</code>.
       */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data.id && !(typeof data.id === 'string' || data.id instanceof String)) {
        throw new Error('Expected the field `id` to be a primitive type in the JSON string but got ' + data.id);
      }
      // ensure the json data is a string
      if (data.sink && !(typeof data.sink === 'string' || data.sink instanceof String)) {
        throw new Error('Expected the field `sink` to be a primitive type in the JSON string but got ' + data.sink);
      }
      // ensure the json data is an array
      if (!Array.isArray(data.types)) {
        throw new Error('Expected the field `types` to be an array in the JSON data but got ' + data.types);
      }
      // validate the optional field `config`
      if (data.config) {
        // data not null
        _WebhooksSubscriptionConfig["default"].validateJSON(data.config);
      }
      return true;
    }
  }]);
  return WebhooksSubscription;
}();
/**
 * Webhooks subscription id
 * @member {String} id
 */
WebhooksSubscription.prototype.id = undefined;

/**
 * Webhooks callback uri.
 * @member {String} sink
 */
WebhooksSubscription.prototype.sink = undefined;

/**
 * [Read Only] True if the webhooks subscription has been verified.
 * @member {Boolean} verified
 */
WebhooksSubscription.prototype.verified = undefined;

/**
 * Webhooks events types.
 * @member {Array.<module:model/EventType>} types
 */
WebhooksSubscription.prototype.types = undefined;

/**
 * @member {module:model/WebhooksSubscriptionConfig} config
 */
WebhooksSubscription.prototype.config = undefined;
var _default = exports["default"] = WebhooksSubscription;
