// @fattureincloud/fattureincloud-js-sdk/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "ArchiveApi", {
  enumerable: true,
  get: function get() {
    return _ArchiveApi["default"];
  }
});
Object.defineProperty(exports, "ArchiveDocument", {
  enumerable: true,
  get: function get() {
    return _ArchiveDocument["default"];
  }
});
Object.defineProperty(exports, "AttachmentData", {
  enumerable: true,
  get: function get() {
    return _AttachmentData["default"];
  }
});
Object.defineProperty(exports, "CashbookApi", {
  enumerable: true,
  get: function get() {
    return _CashbookApi["default"];
  }
});
Object.defineProperty(exports, "CashbookEntry", {
  enumerable: true,
  get: function get() {
    return _CashbookEntry["default"];
  }
});
Object.defineProperty(exports, "CashbookEntryDocument", {
  enumerable: true,
  get: function get() {
    return _CashbookEntryDocument["default"];
  }
});
Object.defineProperty(exports, "CashbookEntryKind", {
  enumerable: true,
  get: function get() {
    return _CashbookEntryKind["default"];
  }
});
Object.defineProperty(exports, "CashbookEntryType", {
  enumerable: true,
  get: function get() {
    return _CashbookEntryType["default"];
  }
});
Object.defineProperty(exports, "City", {
  enumerable: true,
  get: function get() {
    return _City["default"];
  }
});
Object.defineProperty(exports, "Client", {
  enumerable: true,
  get: function get() {
    return _Client["default"];
  }
});
Object.defineProperty(exports, "ClientType", {
  enumerable: true,
  get: function get() {
    return _ClientType["default"];
  }
});
Object.defineProperty(exports, "ClientsApi", {
  enumerable: true,
  get: function get() {
    return _ClientsApi["default"];
  }
});
Object.defineProperty(exports, "CompaniesApi", {
  enumerable: true,
  get: function get() {
    return _CompaniesApi["default"];
  }
});
Object.defineProperty(exports, "Company", {
  enumerable: true,
  get: function get() {
    return _Company["default"];
  }
});
Object.defineProperty(exports, "CompanyInfo", {
  enumerable: true,
  get: function get() {
    return _CompanyInfo["default"];
  }
});
Object.defineProperty(exports, "CompanyInfoAccessInfo", {
  enumerable: true,
  get: function get() {
    return _CompanyInfoAccessInfo["default"];
  }
});
Object.defineProperty(exports, "CompanyInfoPlanInfo", {
  enumerable: true,
  get: function get() {
    return _CompanyInfoPlanInfo["default"];
  }
});
Object.defineProperty(exports, "CompanyInfoPlanInfoFunctions", {
  enumerable: true,
  get: function get() {
    return _CompanyInfoPlanInfoFunctions["default"];
  }
});
Object.defineProperty(exports, "CompanyInfoPlanInfoFunctionsStatus", {
  enumerable: true,
  get: function get() {
    return _CompanyInfoPlanInfoFunctionsStatus["default"];
  }
});
Object.defineProperty(exports, "CompanyInfoPlanInfoLimits", {
  enumerable: true,
  get: function get() {
    return _CompanyInfoPlanInfoLimits["default"];
  }
});
Object.defineProperty(exports, "CompanyPlanUsage", {
  enumerable: true,
  get: function get() {
    return _CompanyPlanUsage["default"];
  }
});
Object.defineProperty(exports, "CompanyType", {
  enumerable: true,
  get: function get() {
    return _CompanyType["default"];
  }
});
Object.defineProperty(exports, "Condition", {
  enumerable: true,
  get: function get() {
    return _Condition["default"];
  }
});
Object.defineProperty(exports, "Conjunction", {
  enumerable: true,
  get: function get() {
    return _Conjunction["default"];
  }
});
Object.defineProperty(exports, "ControlledCompany", {
  enumerable: true,
  get: function get() {
    return _ControlledCompany["default"];
  }
});
Object.defineProperty(exports, "CreateArchiveDocumentRequest", {
  enumerable: true,
  get: function get() {
    return _CreateArchiveDocumentRequest["default"];
  }
});
Object.defineProperty(exports, "CreateArchiveDocumentResponse", {
  enumerable: true,
  get: function get() {
    return _CreateArchiveDocumentResponse["default"];
  }
});
Object.defineProperty(exports, "CreateCashbookEntryRequest", {
  enumerable: true,
  get: function get() {
    return _CreateCashbookEntryRequest["default"];
  }
});
Object.defineProperty(exports, "CreateCashbookEntryResponse", {
  enumerable: true,
  get: function get() {
    return _CreateCashbookEntryResponse["default"];
  }
});
Object.defineProperty(exports, "CreateClientRequest", {
  enumerable: true,
  get: function get() {
    return _CreateClientRequest["default"];
  }
});
Object.defineProperty(exports, "CreateClientResponse", {
  enumerable: true,
  get: function get() {
    return _CreateClientResponse["default"];
  }
});
Object.defineProperty(exports, "CreateF24Request", {
  enumerable: true,
  get: function get() {
    return _CreateF24Request["default"];
  }
});
Object.defineProperty(exports, "CreateF24Response", {
  enumerable: true,
  get: function get() {
    return _CreateF24Response["default"];
  }
});
Object.defineProperty(exports, "CreateIssuedDocumentRequest", {
  enumerable: true,
  get: function get() {
    return _CreateIssuedDocumentRequest["default"];
  }
});
Object.defineProperty(exports, "CreateIssuedDocumentResponse", {
  enumerable: true,
  get: function get() {
    return _CreateIssuedDocumentResponse["default"];
  }
});
Object.defineProperty(exports, "CreatePaymentAccountRequest", {
  enumerable: true,
  get: function get() {
    return _CreatePaymentAccountRequest["default"];
  }
});
Object.defineProperty(exports, "CreatePaymentAccountResponse", {
  enumerable: true,
  get: function get() {
    return _CreatePaymentAccountResponse["default"];
  }
});
Object.defineProperty(exports, "CreatePaymentMethodRequest", {
  enumerable: true,
  get: function get() {
    return _CreatePaymentMethodRequest["default"];
  }
});
Object.defineProperty(exports, "CreatePaymentMethodResponse", {
  enumerable: true,
  get: function get() {
    return _CreatePaymentMethodResponse["default"];
  }
});
Object.defineProperty(exports, "CreateProductRequest", {
  enumerable: true,
  get: function get() {
    return _CreateProductRequest["default"];
  }
});
Object.defineProperty(exports, "CreateProductResponse", {
  enumerable: true,
  get: function get() {
    return _CreateProductResponse["default"];
  }
});
Object.defineProperty(exports, "CreateReceiptRequest", {
  enumerable: true,
  get: function get() {
    return _CreateReceiptRequest["default"];
  }
});
Object.defineProperty(exports, "CreateReceiptResponse", {
  enumerable: true,
  get: function get() {
    return _CreateReceiptResponse["default"];
  }
});
Object.defineProperty(exports, "CreateReceivedDocumentRequest", {
  enumerable: true,
  get: function get() {
    return _CreateReceivedDocumentRequest["default"];
  }
});
Object.defineProperty(exports, "CreateReceivedDocumentResponse", {
  enumerable: true,
  get: function get() {
    return _CreateReceivedDocumentResponse["default"];
  }
});
Object.defineProperty(exports, "CreateSupplierRequest", {
  enumerable: true,
  get: function get() {
    return _CreateSupplierRequest["default"];
  }
});
Object.defineProperty(exports, "CreateSupplierResponse", {
  enumerable: true,
  get: function get() {
    return _CreateSupplierResponse["default"];
  }
});
Object.defineProperty(exports, "CreateVatTypeRequest", {
  enumerable: true,
  get: function get() {
    return _CreateVatTypeRequest["default"];
  }
});
Object.defineProperty(exports, "CreateVatTypeResponse", {
  enumerable: true,
  get: function get() {
    return _CreateVatTypeResponse["default"];
  }
});
Object.defineProperty(exports, "CreateWebhooksSubscriptionRequest", {
  enumerable: true,
  get: function get() {
    return _CreateWebhooksSubscriptionRequest["default"];
  }
});
Object.defineProperty(exports, "CreateWebhooksSubscriptionResponse", {
  enumerable: true,
  get: function get() {
    return _CreateWebhooksSubscriptionResponse["default"];
  }
});
Object.defineProperty(exports, "Currency", {
  enumerable: true,
  get: function get() {
    return _Currency["default"];
  }
});
Object.defineProperty(exports, "DetailedCountry", {
  enumerable: true,
  get: function get() {
    return _DetailedCountry["default"];
  }
});
Object.defineProperty(exports, "Disjunction", {
  enumerable: true,
  get: function get() {
    return _Disjunction["default"];
  }
});
Object.defineProperty(exports, "DocumentTemplate", {
  enumerable: true,
  get: function get() {
    return _DocumentTemplate["default"];
  }
});
Object.defineProperty(exports, "EInvoiceRejectionReason", {
  enumerable: true,
  get: function get() {
    return _EInvoiceRejectionReason["default"];
  }
});
Object.defineProperty(exports, "Email", {
  enumerable: true,
  get: function get() {
    return _Email["default"];
  }
});
Object.defineProperty(exports, "EmailAttachment", {
  enumerable: true,
  get: function get() {
    return _EmailAttachment["default"];
  }
});
Object.defineProperty(exports, "EmailData", {
  enumerable: true,
  get: function get() {
    return _EmailData["default"];
  }
});
Object.defineProperty(exports, "EmailDataDefaultSenderEmail", {
  enumerable: true,
  get: function get() {
    return _EmailDataDefaultSenderEmail["default"];
  }
});
Object.defineProperty(exports, "EmailRecipientStatus", {
  enumerable: true,
  get: function get() {
    return _EmailRecipientStatus["default"];
  }
});
Object.defineProperty(exports, "EmailSchedule", {
  enumerable: true,
  get: function get() {
    return _EmailSchedule["default"];
  }
});
Object.defineProperty(exports, "EmailScheduleInclude", {
  enumerable: true,
  get: function get() {
    return _EmailScheduleInclude["default"];
  }
});
Object.defineProperty(exports, "EmailStatus", {
  enumerable: true,
  get: function get() {
    return _EmailStatus["default"];
  }
});
Object.defineProperty(exports, "EmailsApi", {
  enumerable: true,
  get: function get() {
    return _EmailsApi["default"];
  }
});
Object.defineProperty(exports, "Entity", {
  enumerable: true,
  get: function get() {
    return _Entity["default"];
  }
});
Object.defineProperty(exports, "EntityType", {
  enumerable: true,
  get: function get() {
    return _EntityType["default"];
  }
});
Object.defineProperty(exports, "EventType", {
  enumerable: true,
  get: function get() {
    return _EventType["default"];
  }
});
Object.defineProperty(exports, "Expression", {
  enumerable: true,
  get: function get() {
    return _Expression["default"];
  }
});
Object.defineProperty(exports, "F24", {
  enumerable: true,
  get: function get() {
    return _F["default"];
  }
});
Object.defineProperty(exports, "F24Status", {
  enumerable: true,
  get: function get() {
    return _F24Status["default"];
  }
});
Object.defineProperty(exports, "FattureInCloudPlanType", {
  enumerable: true,
  get: function get() {
    return _FattureInCloudPlanType["default"];
  }
});
Object.defineProperty(exports, "Filter", {
  enumerable: true,
  get: function get() {
    return _Filter["default"];
  }
});
Object.defineProperty(exports, "FunctionStatus", {
  enumerable: true,
  get: function get() {
    return _FunctionStatus["default"];
  }
});
Object.defineProperty(exports, "GetArchiveDocumentResponse", {
  enumerable: true,
  get: function get() {
    return _GetArchiveDocumentResponse["default"];
  }
});
Object.defineProperty(exports, "GetCashbookEntryResponse", {
  enumerable: true,
  get: function get() {
    return _GetCashbookEntryResponse["default"];
  }
});
Object.defineProperty(exports, "GetClientResponse", {
  enumerable: true,
  get: function get() {
    return _GetClientResponse["default"];
  }
});
Object.defineProperty(exports, "GetCompanyInfoResponse", {
  enumerable: true,
  get: function get() {
    return _GetCompanyInfoResponse["default"];
  }
});
Object.defineProperty(exports, "GetCompanyPlanUsageResponse", {
  enumerable: true,
  get: function get() {
    return _GetCompanyPlanUsageResponse["default"];
  }
});
Object.defineProperty(exports, "GetEInvoiceRejectionReasonResponse", {
  enumerable: true,
  get: function get() {
    return _GetEInvoiceRejectionReasonResponse["default"];
  }
});
Object.defineProperty(exports, "GetEmailDataResponse", {
  enumerable: true,
  get: function get() {
    return _GetEmailDataResponse["default"];
  }
});
Object.defineProperty(exports, "GetExistingIssuedDocumentTotalsRequest", {
  enumerable: true,
  get: function get() {
    return _GetExistingIssuedDocumentTotalsRequest["default"];
  }
});
Object.defineProperty(exports, "GetExistingIssuedDocumentTotalsResponse", {
  enumerable: true,
  get: function get() {
    return _GetExistingIssuedDocumentTotalsResponse["default"];
  }
});
Object.defineProperty(exports, "GetExistingReceivedDocumentTotalsRequest", {
  enumerable: true,
  get: function get() {
    return _GetExistingReceivedDocumentTotalsRequest["default"];
  }
});
Object.defineProperty(exports, "GetExistingReceivedDocumentTotalsResponse", {
  enumerable: true,
  get: function get() {
    return _GetExistingReceivedDocumentTotalsResponse["default"];
  }
});
Object.defineProperty(exports, "GetF24Response", {
  enumerable: true,
  get: function get() {
    return _GetF24Response["default"];
  }
});
Object.defineProperty(exports, "GetIssuedDocumentPreCreateInfoResponse", {
  enumerable: true,
  get: function get() {
    return _GetIssuedDocumentPreCreateInfoResponse["default"];
  }
});
Object.defineProperty(exports, "GetIssuedDocumentResponse", {
  enumerable: true,
  get: function get() {
    return _GetIssuedDocumentResponse["default"];
  }
});
Object.defineProperty(exports, "GetNewIssuedDocumentTotalsRequest", {
  enumerable: true,
  get: function get() {
    return _GetNewIssuedDocumentTotalsRequest["default"];
  }
});
Object.defineProperty(exports, "GetNewIssuedDocumentTotalsResponse", {
  enumerable: true,
  get: function get() {
    return _GetNewIssuedDocumentTotalsResponse["default"];
  }
});
Object.defineProperty(exports, "GetNewReceivedDocumentTotalsRequest", {
  enumerable: true,
  get: function get() {
    return _GetNewReceivedDocumentTotalsRequest["default"];
  }
});
Object.defineProperty(exports, "GetNewReceivedDocumentTotalsResponse", {
  enumerable: true,
  get: function get() {
    return _GetNewReceivedDocumentTotalsResponse["default"];
  }
});
Object.defineProperty(exports, "GetPaymentAccountResponse", {
  enumerable: true,
  get: function get() {
    return _GetPaymentAccountResponse["default"];
  }
});
Object.defineProperty(exports, "GetPaymentMethodResponse", {
  enumerable: true,
  get: function get() {
    return _GetPaymentMethodResponse["default"];
  }
});
Object.defineProperty(exports, "GetProductResponse", {
  enumerable: true,
  get: function get() {
    return _GetProductResponse["default"];
  }
});
Object.defineProperty(exports, "GetReceiptPreCreateInfoResponse", {
  enumerable: true,
  get: function get() {
    return _GetReceiptPreCreateInfoResponse["default"];
  }
});
Object.defineProperty(exports, "GetReceiptResponse", {
  enumerable: true,
  get: function get() {
    return _GetReceiptResponse["default"];
  }
});
Object.defineProperty(exports, "GetReceiptsMonthlyTotalsResponse", {
  enumerable: true,
  get: function get() {
    return _GetReceiptsMonthlyTotalsResponse["default"];
  }
});
Object.defineProperty(exports, "GetReceivedDocumentPreCreateInfoResponse", {
  enumerable: true,
  get: function get() {
    return _GetReceivedDocumentPreCreateInfoResponse["default"];
  }
});
Object.defineProperty(exports, "GetReceivedDocumentResponse", {
  enumerable: true,
  get: function get() {
    return _GetReceivedDocumentResponse["default"];
  }
});
Object.defineProperty(exports, "GetSupplierResponse", {
  enumerable: true,
  get: function get() {
    return _GetSupplierResponse["default"];
  }
});
Object.defineProperty(exports, "GetUserInfoResponse", {
  enumerable: true,
  get: function get() {
    return _GetUserInfoResponse["default"];
  }
});
Object.defineProperty(exports, "GetUserInfoResponseEmailConfirmationState", {
  enumerable: true,
  get: function get() {
    return _GetUserInfoResponseEmailConfirmationState["default"];
  }
});
Object.defineProperty(exports, "GetUserInfoResponseInfo", {
  enumerable: true,
  get: function get() {
    return _GetUserInfoResponseInfo["default"];
  }
});
Object.defineProperty(exports, "GetVatTypeResponse", {
  enumerable: true,
  get: function get() {
    return _GetVatTypeResponse["default"];
  }
});
Object.defineProperty(exports, "GetWebhooksSubscriptionResponse", {
  enumerable: true,
  get: function get() {
    return _GetWebhooksSubscriptionResponse["default"];
  }
});
Object.defineProperty(exports, "InfoApi", {
  enumerable: true,
  get: function get() {
    return _InfoApi["default"];
  }
});
Object.defineProperty(exports, "IssuedDocument", {
  enumerable: true,
  get: function get() {
    return _IssuedDocument["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentEiData", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentEiData["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentExtraData", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentExtraData["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentItemsListItem", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentItemsListItem["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentOptions", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentOptions["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentPaymentsListItem", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentPaymentsListItem["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentPaymentsListItemPaymentTerms", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentPaymentsListItemPaymentTerms["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentPreCreateInfo", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentPreCreateInfo["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentPreCreateInfoDefaultValues", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentPreCreateInfoDefaultValues["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentPreCreateInfoExtraDataDefaultValues", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentPreCreateInfoExtraDataDefaultValues["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentPreCreateInfoItemsDefaultValues", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentPreCreateInfoItemsDefaultValues["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentStatus", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentStatus["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentTotals", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentTotals["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentType", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentType["default"];
  }
});
Object.defineProperty(exports, "IssuedDocumentsApi", {
  enumerable: true,
  get: function get() {
    return _IssuedDocumentsApi["default"];
  }
});
Object.defineProperty(exports, "IssuedEInvoicesApi", {
  enumerable: true,
  get: function get() {
    return _IssuedEInvoicesApi["default"];
  }
});
Object.defineProperty(exports, "JoinIssuedDocumentsResponse", {
  enumerable: true,
  get: function get() {
    return _JoinIssuedDocumentsResponse["default"];
  }
});
Object.defineProperty(exports, "Language", {
  enumerable: true,
  get: function get() {
    return _Language["default"];
  }
});
Object.defineProperty(exports, "ListArchiveCategoriesResponse", {
  enumerable: true,
  get: function get() {
    return _ListArchiveCategoriesResponse["default"];
  }
});
Object.defineProperty(exports, "ListArchiveDocumentsResponse", {
  enumerable: true,
  get: function get() {
    return _ListArchiveDocumentsResponse["default"];
  }
});
Object.defineProperty(exports, "ListArchiveDocumentsResponsePage", {
  enumerable: true,
  get: function get() {
    return _ListArchiveDocumentsResponsePage["default"];
  }
});
Object.defineProperty(exports, "ListCashbookEntriesResponse", {
  enumerable: true,
  get: function get() {
    return _ListCashbookEntriesResponse["default"];
  }
});
Object.defineProperty(exports, "ListCitiesResponse", {
  enumerable: true,
  get: function get() {
    return _ListCitiesResponse["default"];
  }
});
Object.defineProperty(exports, "ListClientsResponse", {
  enumerable: true,
  get: function get() {
    return _ListClientsResponse["default"];
  }
});
Object.defineProperty(exports, "ListClientsResponsePage", {
  enumerable: true,
  get: function get() {
    return _ListClientsResponsePage["default"];
  }
});
Object.defineProperty(exports, "ListCostCentersResponse", {
  enumerable: true,
  get: function get() {
    return _ListCostCentersResponse["default"];
  }
});
Object.defineProperty(exports, "ListCountriesResponse", {
  enumerable: true,
  get: function get() {
    return _ListCountriesResponse["default"];
  }
});
Object.defineProperty(exports, "ListCurrenciesResponse", {
  enumerable: true,
  get: function get() {
    return _ListCurrenciesResponse["default"];
  }
});
Object.defineProperty(exports, "ListDeliveryNotesDefaultCausalsResponse", {
  enumerable: true,
  get: function get() {
    return _ListDeliveryNotesDefaultCausalsResponse["default"];
  }
});
Object.defineProperty(exports, "ListDetailedCountriesResponse", {
  enumerable: true,
  get: function get() {
    return _ListDetailedCountriesResponse["default"];
  }
});
Object.defineProperty(exports, "ListEmailsResponse", {
  enumerable: true,
  get: function get() {
    return _ListEmailsResponse["default"];
  }
});
Object.defineProperty(exports, "ListEmailsResponsePage", {
  enumerable: true,
  get: function get() {
    return _ListEmailsResponsePage["default"];
  }
});
Object.defineProperty(exports, "ListF24Response", {
  enumerable: true,
  get: function get() {
    return _ListF24Response["default"];
  }
});
Object.defineProperty(exports, "ListF24ResponseAggregatedData", {
  enumerable: true,
  get: function get() {
    return _ListF24ResponseAggregatedData["default"];
  }
});
Object.defineProperty(exports, "ListF24ResponseAggregation", {
  enumerable: true,
  get: function get() {
    return _ListF24ResponseAggregation["default"];
  }
});
Object.defineProperty(exports, "ListF24ResponsePage", {
  enumerable: true,
  get: function get() {
    return _ListF24ResponsePage["default"];
  }
});
Object.defineProperty(exports, "ListIssuedDocumentsResponse", {
  enumerable: true,
  get: function get() {
    return _ListIssuedDocumentsResponse["default"];
  }
});
Object.defineProperty(exports, "ListIssuedDocumentsResponsePage", {
  enumerable: true,
  get: function get() {
    return _ListIssuedDocumentsResponsePage["default"];
  }
});
Object.defineProperty(exports, "ListLanguagesResponse", {
  enumerable: true,
  get: function get() {
    return _ListLanguagesResponse["default"];
  }
});
Object.defineProperty(exports, "ListPaymentAccountsResponse", {
  enumerable: true,
  get: function get() {
    return _ListPaymentAccountsResponse["default"];
  }
});
Object.defineProperty(exports, "ListPaymentMethodsResponse", {
  enumerable: true,
  get: function get() {
    return _ListPaymentMethodsResponse["default"];
  }
});
Object.defineProperty(exports, "ListProductCategoriesResponse", {
  enumerable: true,
  get: function get() {
    return _ListProductCategoriesResponse["default"];
  }
});
Object.defineProperty(exports, "ListProductsResponse", {
  enumerable: true,
  get: function get() {
    return _ListProductsResponse["default"];
  }
});
Object.defineProperty(exports, "ListProductsResponsePage", {
  enumerable: true,
  get: function get() {
    return _ListProductsResponsePage["default"];
  }
});
Object.defineProperty(exports, "ListReceiptsResponse", {
  enumerable: true,
  get: function get() {
    return _ListReceiptsResponse["default"];
  }
});
Object.defineProperty(exports, "ListReceiptsResponsePage", {
  enumerable: true,
  get: function get() {
    return _ListReceiptsResponsePage["default"];
  }
});
Object.defineProperty(exports, "ListReceivedDocumentCategoriesResponse", {
  enumerable: true,
  get: function get() {
    return _ListReceivedDocumentCategoriesResponse["default"];
  }
});
Object.defineProperty(exports, "ListReceivedDocumentsResponse", {
  enumerable: true,
  get: function get() {
    return _ListReceivedDocumentsResponse["default"];
  }
});
Object.defineProperty(exports, "ListReceivedDocumentsResponsePage", {
  enumerable: true,
  get: function get() {
    return _ListReceivedDocumentsResponsePage["default"];
  }
});
Object.defineProperty(exports, "ListRevenueCentersResponse", {
  enumerable: true,
  get: function get() {
    return _ListRevenueCentersResponse["default"];
  }
});
Object.defineProperty(exports, "ListSuppliersResponse", {
  enumerable: true,
  get: function get() {
    return _ListSuppliersResponse["default"];
  }
});
Object.defineProperty(exports, "ListSuppliersResponsePage", {
  enumerable: true,
  get: function get() {
    return _ListSuppliersResponsePage["default"];
  }
});
Object.defineProperty(exports, "ListTemplatesResponse", {
  enumerable: true,
  get: function get() {
    return _ListTemplatesResponse["default"];
  }
});
Object.defineProperty(exports, "ListUnitsOfMeasureResponse", {
  enumerable: true,
  get: function get() {
    return _ListUnitsOfMeasureResponse["default"];
  }
});
Object.defineProperty(exports, "ListUserCompaniesResponse", {
  enumerable: true,
  get: function get() {
    return _ListUserCompaniesResponse["default"];
  }
});
Object.defineProperty(exports, "ListUserCompaniesResponseData", {
  enumerable: true,
  get: function get() {
    return _ListUserCompaniesResponseData["default"];
  }
});
Object.defineProperty(exports, "ListVatTypesResponse", {
  enumerable: true,
  get: function get() {
    return _ListVatTypesResponse["default"];
  }
});
Object.defineProperty(exports, "ListWebhooksSubscriptionsResponse", {
  enumerable: true,
  get: function get() {
    return _ListWebhooksSubscriptionsResponse["default"];
  }
});
Object.defineProperty(exports, "ModifyArchiveDocumentRequest", {
  enumerable: true,
  get: function get() {
    return _ModifyArchiveDocumentRequest["default"];
  }
});
Object.defineProperty(exports, "ModifyArchiveDocumentResponse", {
  enumerable: true,
  get: function get() {
    return _ModifyArchiveDocumentResponse["default"];
  }
});
Object.defineProperty(exports, "ModifyCashbookEntryRequest", {
  enumerable: true,
  get: function get() {
    return _ModifyCashbookEntryRequest["default"];
  }
});
Object.defineProperty(exports, "ModifyCashbookEntryResponse", {
  enumerable: true,
  get: function get() {
    return _ModifyCashbookEntryResponse["default"];
  }
});
Object.defineProperty(exports, "ModifyClientRequest", {
  enumerable: true,
  get: function get() {
    return _ModifyClientRequest["default"];
  }
});
Object.defineProperty(exports, "ModifyClientResponse", {
  enumerable: true,
  get: function get() {
    return _ModifyClientResponse["default"];
  }
});
Object.defineProperty(exports, "ModifyF24Request", {
  enumerable: true,
  get: function get() {
    return _ModifyF24Request["default"];
  }
});
Object.defineProperty(exports, "ModifyF24Response", {
  enumerable: true,
  get: function get() {
    return _ModifyF24Response["default"];
  }
});
Object.defineProperty(exports, "ModifyIssuedDocumentRequest", {
  enumerable: true,
  get: function get() {
    return _ModifyIssuedDocumentRequest["default"];
  }
});
Object.defineProperty(exports, "ModifyIssuedDocumentResponse", {
  enumerable: true,
  get: function get() {
    return _ModifyIssuedDocumentResponse["default"];
  }
});
Object.defineProperty(exports, "ModifyPaymentAccountRequest", {
  enumerable: true,
  get: function get() {
    return _ModifyPaymentAccountRequest["default"];
  }
});
Object.defineProperty(exports, "ModifyPaymentAccountResponse", {
  enumerable: true,
  get: function get() {
    return _ModifyPaymentAccountResponse["default"];
  }
});
Object.defineProperty(exports, "ModifyPaymentMethodRequest", {
  enumerable: true,
  get: function get() {
    return _ModifyPaymentMethodRequest["default"];
  }
});
Object.defineProperty(exports, "ModifyPaymentMethodResponse", {
  enumerable: true,
  get: function get() {
    return _ModifyPaymentMethodResponse["default"];
  }
});
Object.defineProperty(exports, "ModifyProductRequest", {
  enumerable: true,
  get: function get() {
    return _ModifyProductRequest["default"];
  }
});
Object.defineProperty(exports, "ModifyProductResponse", {
  enumerable: true,
  get: function get() {
    return _ModifyProductResponse["default"];
  }
});
Object.defineProperty(exports, "ModifyReceiptRequest", {
  enumerable: true,
  get: function get() {
    return _ModifyReceiptRequest["default"];
  }
});
Object.defineProperty(exports, "ModifyReceiptResponse", {
  enumerable: true,
  get: function get() {
    return _ModifyReceiptResponse["default"];
  }
});
Object.defineProperty(exports, "ModifyReceivedDocumentRequest", {
  enumerable: true,
  get: function get() {
    return _ModifyReceivedDocumentRequest["default"];
  }
});
Object.defineProperty(exports, "ModifyReceivedDocumentResponse", {
  enumerable: true,
  get: function get() {
    return _ModifyReceivedDocumentResponse["default"];
  }
});
Object.defineProperty(exports, "ModifySupplierRequest", {
  enumerable: true,
  get: function get() {
    return _ModifySupplierRequest["default"];
  }
});
Object.defineProperty(exports, "ModifySupplierResponse", {
  enumerable: true,
  get: function get() {
    return _ModifySupplierResponse["default"];
  }
});
Object.defineProperty(exports, "ModifyVatTypeRequest", {
  enumerable: true,
  get: function get() {
    return _ModifyVatTypeRequest["default"];
  }
});
Object.defineProperty(exports, "ModifyVatTypeResponse", {
  enumerable: true,
  get: function get() {
    return _ModifyVatTypeResponse["default"];
  }
});
Object.defineProperty(exports, "ModifyWebhooksSubscriptionRequest", {
  enumerable: true,
  get: function get() {
    return _ModifyWebhooksSubscriptionRequest["default"];
  }
});
Object.defineProperty(exports, "ModifyWebhooksSubscriptionResponse", {
  enumerable: true,
  get: function get() {
    return _ModifyWebhooksSubscriptionResponse["default"];
  }
});
Object.defineProperty(exports, "MonthlyTotal", {
  enumerable: true,
  get: function get() {
    return _MonthlyTotal["default"];
  }
});
Object.defineProperty(exports, "OAuth2AuthorizationCodeManager", {
  enumerable: true,
  get: function get() {
    return _OAuth["default"];
  }
});
Object.defineProperty(exports, "OAuth2AuthorizationCodeParams", {
  enumerable: true,
  get: function get() {
    return _OAuth2AuthorizationCodeParams["default"];
  }
});
Object.defineProperty(exports, "OAuth2AuthorizationCodeTokenResponse", {
  enumerable: true,
  get: function get() {
    return _OAuth2AuthorizationCodeTokenResponse["default"];
  }
});
Object.defineProperty(exports, "Operator", {
  enumerable: true,
  get: function get() {
    return _Operator["default"];
  }
});
Object.defineProperty(exports, "OriginalDocumentType", {
  enumerable: true,
  get: function get() {
    return _OriginalDocumentType["default"];
  }
});
Object.defineProperty(exports, "Pagination", {
  enumerable: true,
  get: function get() {
    return _Pagination["default"];
  }
});
Object.defineProperty(exports, "PaymentAccount", {
  enumerable: true,
  get: function get() {
    return _PaymentAccount["default"];
  }
});
Object.defineProperty(exports, "PaymentAccountType", {
  enumerable: true,
  get: function get() {
    return _PaymentAccountType["default"];
  }
});
Object.defineProperty(exports, "PaymentMethod", {
  enumerable: true,
  get: function get() {
    return _PaymentMethod["default"];
  }
});
Object.defineProperty(exports, "PaymentMethodDetails", {
  enumerable: true,
  get: function get() {
    return _PaymentMethodDetails["default"];
  }
});
Object.defineProperty(exports, "PaymentMethodType", {
  enumerable: true,
  get: function get() {
    return _PaymentMethodType["default"];
  }
});
Object.defineProperty(exports, "PaymentTermsType", {
  enumerable: true,
  get: function get() {
    return _PaymentTermsType["default"];
  }
});
Object.defineProperty(exports, "PermissionLevel", {
  enumerable: true,
  get: function get() {
    return _PermissionLevel["default"];
  }
});
Object.defineProperty(exports, "Permissions", {
  enumerable: true,
  get: function get() {
    return _Permissions["default"];
  }
});
Object.defineProperty(exports, "PermissionsFicIssuedDocumentsDetailed", {
  enumerable: true,
  get: function get() {
    return _PermissionsFicIssuedDocumentsDetailed["default"];
  }
});
Object.defineProperty(exports, "Product", {
  enumerable: true,
  get: function get() {
    return _Product["default"];
  }
});
Object.defineProperty(exports, "ProductsApi", {
  enumerable: true,
  get: function get() {
    return _ProductsApi["default"];
  }
});
Object.defineProperty(exports, "Receipt", {
  enumerable: true,
  get: function get() {
    return _Receipt["default"];
  }
});
Object.defineProperty(exports, "ReceiptItemsListItem", {
  enumerable: true,
  get: function get() {
    return _ReceiptItemsListItem["default"];
  }
});
Object.defineProperty(exports, "ReceiptPreCreateInfo", {
  enumerable: true,
  get: function get() {
    return _ReceiptPreCreateInfo["default"];
  }
});
Object.defineProperty(exports, "ReceiptType", {
  enumerable: true,
  get: function get() {
    return _ReceiptType["default"];
  }
});
Object.defineProperty(exports, "ReceiptsApi", {
  enumerable: true,
  get: function get() {
    return _ReceiptsApi["default"];
  }
});
Object.defineProperty(exports, "ReceivedDocument", {
  enumerable: true,
  get: function get() {
    return _ReceivedDocument["default"];
  }
});
Object.defineProperty(exports, "ReceivedDocumentInfo", {
  enumerable: true,
  get: function get() {
    return _ReceivedDocumentInfo["default"];
  }
});
Object.defineProperty(exports, "ReceivedDocumentInfoDefaultValues", {
  enumerable: true,
  get: function get() {
    return _ReceivedDocumentInfoDefaultValues["default"];
  }
});
Object.defineProperty(exports, "ReceivedDocumentInfoItemsDefaultValues", {
  enumerable: true,
  get: function get() {
    return _ReceivedDocumentInfoItemsDefaultValues["default"];
  }
});
Object.defineProperty(exports, "ReceivedDocumentItemsListItem", {
  enumerable: true,
  get: function get() {
    return _ReceivedDocumentItemsListItem["default"];
  }
});
Object.defineProperty(exports, "ReceivedDocumentPaymentsListItem", {
  enumerable: true,
  get: function get() {
    return _ReceivedDocumentPaymentsListItem["default"];
  }
});
Object.defineProperty(exports, "ReceivedDocumentPaymentsListItemPaymentTerms", {
  enumerable: true,
  get: function get() {
    return _ReceivedDocumentPaymentsListItemPaymentTerms["default"];
  }
});
Object.defineProperty(exports, "ReceivedDocumentTotals", {
  enumerable: true,
  get: function get() {
    return _ReceivedDocumentTotals["default"];
  }
});
Object.defineProperty(exports, "ReceivedDocumentType", {
  enumerable: true,
  get: function get() {
    return _ReceivedDocumentType["default"];
  }
});
Object.defineProperty(exports, "ReceivedDocumentsApi", {
  enumerable: true,
  get: function get() {
    return _ReceivedDocumentsApi["default"];
  }
});
Object.defineProperty(exports, "ScheduleEmailRequest", {
  enumerable: true,
  get: function get() {
    return _ScheduleEmailRequest["default"];
  }
});
Object.defineProperty(exports, "Scope", {
  enumerable: true,
  get: function get() {
    return _Scope["default"];
  }
});
Object.defineProperty(exports, "SendEInvoiceRequest", {
  enumerable: true,
  get: function get() {
    return _SendEInvoiceRequest["default"];
  }
});
Object.defineProperty(exports, "SendEInvoiceRequestData", {
  enumerable: true,
  get: function get() {
    return _SendEInvoiceRequestData["default"];
  }
});
Object.defineProperty(exports, "SendEInvoiceRequestOptions", {
  enumerable: true,
  get: function get() {
    return _SendEInvoiceRequestOptions["default"];
  }
});
Object.defineProperty(exports, "SendEInvoiceResponse", {
  enumerable: true,
  get: function get() {
    return _SendEInvoiceResponse["default"];
  }
});
Object.defineProperty(exports, "SendEInvoiceResponseData", {
  enumerable: true,
  get: function get() {
    return _SendEInvoiceResponseData["default"];
  }
});
Object.defineProperty(exports, "SenderEmail", {
  enumerable: true,
  get: function get() {
    return _SenderEmail["default"];
  }
});
Object.defineProperty(exports, "SettingsApi", {
  enumerable: true,
  get: function get() {
    return _SettingsApi["default"];
  }
});
Object.defineProperty(exports, "ShowTotalsMode", {
  enumerable: true,
  get: function get() {
    return _ShowTotalsMode["default"];
  }
});
Object.defineProperty(exports, "Supplier", {
  enumerable: true,
  get: function get() {
    return _Supplier["default"];
  }
});
Object.defineProperty(exports, "SupplierType", {
  enumerable: true,
  get: function get() {
    return _SupplierType["default"];
  }
});
Object.defineProperty(exports, "SuppliersApi", {
  enumerable: true,
  get: function get() {
    return _SuppliersApi["default"];
  }
});
Object.defineProperty(exports, "TaxesApi", {
  enumerable: true,
  get: function get() {
    return _TaxesApi["default"];
  }
});
Object.defineProperty(exports, "TransformIssuedDocumentResponse", {
  enumerable: true,
  get: function get() {
    return _TransformIssuedDocumentResponse["default"];
  }
});
Object.defineProperty(exports, "UploadArchiveAttachmentResponse", {
  enumerable: true,
  get: function get() {
    return _UploadArchiveAttachmentResponse["default"];
  }
});
Object.defineProperty(exports, "UploadF24AttachmentResponse", {
  enumerable: true,
  get: function get() {
    return _UploadF24AttachmentResponse["default"];
  }
});
Object.defineProperty(exports, "UploadIssuedDocumentAttachmentResponse", {
  enumerable: true,
  get: function get() {
    return _UploadIssuedDocumentAttachmentResponse["default"];
  }
});
Object.defineProperty(exports, "UploadReceivedDocumentAttachmentResponse", {
  enumerable: true,
  get: function get() {
    return _UploadReceivedDocumentAttachmentResponse["default"];
  }
});
Object.defineProperty(exports, "User", {
  enumerable: true,
  get: function get() {
    return _User["default"];
  }
});
Object.defineProperty(exports, "UserApi", {
  enumerable: true,
  get: function get() {
    return _UserApi["default"];
  }
});
Object.defineProperty(exports, "UserCompanyRole", {
  enumerable: true,
  get: function get() {
    return _UserCompanyRole["default"];
  }
});
Object.defineProperty(exports, "VatItem", {
  enumerable: true,
  get: function get() {
    return _VatItem["default"];
  }
});
Object.defineProperty(exports, "VatKind", {
  enumerable: true,
  get: function get() {
    return _VatKind["default"];
  }
});
Object.defineProperty(exports, "VatType", {
  enumerable: true,
  get: function get() {
    return _VatType["default"];
  }
});
Object.defineProperty(exports, "VerifyEInvoiceXmlErrorResponse", {
  enumerable: true,
  get: function get() {
    return _VerifyEInvoiceXmlErrorResponse["default"];
  }
});
Object.defineProperty(exports, "VerifyEInvoiceXmlErrorResponseError", {
  enumerable: true,
  get: function get() {
    return _VerifyEInvoiceXmlErrorResponseError["default"];
  }
});
Object.defineProperty(exports, "VerifyEInvoiceXmlErrorResponseErrorValidationResult", {
  enumerable: true,
  get: function get() {
    return _VerifyEInvoiceXmlErrorResponseErrorValidationResult["default"];
  }
});
Object.defineProperty(exports, "VerifyEInvoiceXmlErrorResponseExtra", {
  enumerable: true,
  get: function get() {
    return _VerifyEInvoiceXmlErrorResponseExtra["default"];
  }
});
Object.defineProperty(exports, "VerifyEInvoiceXmlResponse", {
  enumerable: true,
  get: function get() {
    return _VerifyEInvoiceXmlResponse["default"];
  }
});
Object.defineProperty(exports, "VerifyEInvoiceXmlResponseData", {
  enumerable: true,
  get: function get() {
    return _VerifyEInvoiceXmlResponseData["default"];
  }
});
Object.defineProperty(exports, "WebhooksApi", {
  enumerable: true,
  get: function get() {
    return _WebhooksApi["default"];
  }
});
Object.defineProperty(exports, "WebhooksSubscription", {
  enumerable: true,
  get: function get() {
    return _WebhooksSubscription["default"];
  }
});
Object.defineProperty(exports, "WebhooksSubscriptionConfig", {
  enumerable: true,
  get: function get() {
    return _WebhooksSubscriptionConfig["default"];
  }
});
Object.defineProperty(exports, "WebhooksSubscriptionMapping", {
  enumerable: true,
  get: function get() {
    return _WebhooksSubscriptionMapping["default"];
  }
});
var _ApiClient = _interopRequireDefault(require("./ApiClient"));
var _ArchiveDocument = _interopRequireDefault(require("./model/ArchiveDocument"));
var _AttachmentData = _interopRequireDefault(require("./model/AttachmentData"));
var _CashbookEntry = _interopRequireDefault(require("./model/CashbookEntry"));
var _CashbookEntryDocument = _interopRequireDefault(require("./model/CashbookEntryDocument"));
var _CashbookEntryKind = _interopRequireDefault(require("./model/CashbookEntryKind"));
var _CashbookEntryType = _interopRequireDefault(require("./model/CashbookEntryType"));
var _City = _interopRequireDefault(require("./model/City"));
var _Client = _interopRequireDefault(require("./model/Client"));
var _ClientType = _interopRequireDefault(require("./model/ClientType"));
var _Company = _interopRequireDefault(require("./model/Company"));
var _CompanyInfo = _interopRequireDefault(require("./model/CompanyInfo"));
var _CompanyInfoAccessInfo = _interopRequireDefault(require("./model/CompanyInfoAccessInfo"));
var _CompanyInfoPlanInfo = _interopRequireDefault(require("./model/CompanyInfoPlanInfo"));
var _CompanyInfoPlanInfoFunctions = _interopRequireDefault(require("./model/CompanyInfoPlanInfoFunctions"));
var _CompanyInfoPlanInfoFunctionsStatus = _interopRequireDefault(require("./model/CompanyInfoPlanInfoFunctionsStatus"));
var _CompanyInfoPlanInfoLimits = _interopRequireDefault(require("./model/CompanyInfoPlanInfoLimits"));
var _CompanyPlanUsage = _interopRequireDefault(require("./model/CompanyPlanUsage"));
var _CompanyType = _interopRequireDefault(require("./model/CompanyType"));
var _ControlledCompany = _interopRequireDefault(require("./model/ControlledCompany"));
var _CreateArchiveDocumentRequest = _interopRequireDefault(require("./model/CreateArchiveDocumentRequest"));
var _CreateArchiveDocumentResponse = _interopRequireDefault(require("./model/CreateArchiveDocumentResponse"));
var _CreateCashbookEntryRequest = _interopRequireDefault(require("./model/CreateCashbookEntryRequest"));
var _CreateCashbookEntryResponse = _interopRequireDefault(require("./model/CreateCashbookEntryResponse"));
var _CreateClientRequest = _interopRequireDefault(require("./model/CreateClientRequest"));
var _CreateClientResponse = _interopRequireDefault(require("./model/CreateClientResponse"));
var _CreateF24Request = _interopRequireDefault(require("./model/CreateF24Request"));
var _CreateF24Response = _interopRequireDefault(require("./model/CreateF24Response"));
var _CreateIssuedDocumentRequest = _interopRequireDefault(require("./model/CreateIssuedDocumentRequest"));
var _CreateIssuedDocumentResponse = _interopRequireDefault(require("./model/CreateIssuedDocumentResponse"));
var _CreatePaymentAccountRequest = _interopRequireDefault(require("./model/CreatePaymentAccountRequest"));
var _CreatePaymentAccountResponse = _interopRequireDefault(require("./model/CreatePaymentAccountResponse"));
var _CreatePaymentMethodRequest = _interopRequireDefault(require("./model/CreatePaymentMethodRequest"));
var _CreatePaymentMethodResponse = _interopRequireDefault(require("./model/CreatePaymentMethodResponse"));
var _CreateProductRequest = _interopRequireDefault(require("./model/CreateProductRequest"));
var _CreateProductResponse = _interopRequireDefault(require("./model/CreateProductResponse"));
var _CreateReceiptRequest = _interopRequireDefault(require("./model/CreateReceiptRequest"));
var _CreateReceiptResponse = _interopRequireDefault(require("./model/CreateReceiptResponse"));
var _CreateReceivedDocumentRequest = _interopRequireDefault(require("./model/CreateReceivedDocumentRequest"));
var _CreateReceivedDocumentResponse = _interopRequireDefault(require("./model/CreateReceivedDocumentResponse"));
var _CreateSupplierRequest = _interopRequireDefault(require("./model/CreateSupplierRequest"));
var _CreateSupplierResponse = _interopRequireDefault(require("./model/CreateSupplierResponse"));
var _CreateVatTypeRequest = _interopRequireDefault(require("./model/CreateVatTypeRequest"));
var _CreateVatTypeResponse = _interopRequireDefault(require("./model/CreateVatTypeResponse"));
var _CreateWebhooksSubscriptionRequest = _interopRequireDefault(require("./model/CreateWebhooksSubscriptionRequest"));
var _CreateWebhooksSubscriptionResponse = _interopRequireDefault(require("./model/CreateWebhooksSubscriptionResponse"));
var _Currency = _interopRequireDefault(require("./model/Currency"));
var _DetailedCountry = _interopRequireDefault(require("./model/DetailedCountry"));
var _DocumentTemplate = _interopRequireDefault(require("./model/DocumentTemplate"));
var _EInvoiceRejectionReason = _interopRequireDefault(require("./model/EInvoiceRejectionReason"));
var _Email = _interopRequireDefault(require("./model/Email"));
var _EmailAttachment = _interopRequireDefault(require("./model/EmailAttachment"));
var _EmailData = _interopRequireDefault(require("./model/EmailData"));
var _EmailDataDefaultSenderEmail = _interopRequireDefault(require("./model/EmailDataDefaultSenderEmail"));
var _EmailRecipientStatus = _interopRequireDefault(require("./model/EmailRecipientStatus"));
var _EmailSchedule = _interopRequireDefault(require("./model/EmailSchedule"));
var _EmailScheduleInclude = _interopRequireDefault(require("./model/EmailScheduleInclude"));
var _EmailStatus = _interopRequireDefault(require("./model/EmailStatus"));
var _Entity = _interopRequireDefault(require("./model/Entity"));
var _EntityType = _interopRequireDefault(require("./model/EntityType"));
var _EventType = _interopRequireDefault(require("./model/EventType"));
var _F = _interopRequireDefault(require("./model/F24"));
var _F24Status = _interopRequireDefault(require("./model/F24Status"));
var _FattureInCloudPlanType = _interopRequireDefault(require("./model/FattureInCloudPlanType"));
var _FunctionStatus = _interopRequireDefault(require("./model/FunctionStatus"));
var _GetArchiveDocumentResponse = _interopRequireDefault(require("./model/GetArchiveDocumentResponse"));
var _GetCashbookEntryResponse = _interopRequireDefault(require("./model/GetCashbookEntryResponse"));
var _GetClientResponse = _interopRequireDefault(require("./model/GetClientResponse"));
var _GetCompanyInfoResponse = _interopRequireDefault(require("./model/GetCompanyInfoResponse"));
var _GetCompanyPlanUsageResponse = _interopRequireDefault(require("./model/GetCompanyPlanUsageResponse"));
var _GetEInvoiceRejectionReasonResponse = _interopRequireDefault(require("./model/GetEInvoiceRejectionReasonResponse"));
var _GetEmailDataResponse = _interopRequireDefault(require("./model/GetEmailDataResponse"));
var _GetExistingIssuedDocumentTotalsRequest = _interopRequireDefault(require("./model/GetExistingIssuedDocumentTotalsRequest"));
var _GetExistingIssuedDocumentTotalsResponse = _interopRequireDefault(require("./model/GetExistingIssuedDocumentTotalsResponse"));
var _GetExistingReceivedDocumentTotalsRequest = _interopRequireDefault(require("./model/GetExistingReceivedDocumentTotalsRequest"));
var _GetExistingReceivedDocumentTotalsResponse = _interopRequireDefault(require("./model/GetExistingReceivedDocumentTotalsResponse"));
var _GetF24Response = _interopRequireDefault(require("./model/GetF24Response"));
var _GetIssuedDocumentPreCreateInfoResponse = _interopRequireDefault(require("./model/GetIssuedDocumentPreCreateInfoResponse"));
var _GetIssuedDocumentResponse = _interopRequireDefault(require("./model/GetIssuedDocumentResponse"));
var _GetNewIssuedDocumentTotalsRequest = _interopRequireDefault(require("./model/GetNewIssuedDocumentTotalsRequest"));
var _GetNewIssuedDocumentTotalsResponse = _interopRequireDefault(require("./model/GetNewIssuedDocumentTotalsResponse"));
var _GetNewReceivedDocumentTotalsRequest = _interopRequireDefault(require("./model/GetNewReceivedDocumentTotalsRequest"));
var _GetNewReceivedDocumentTotalsResponse = _interopRequireDefault(require("./model/GetNewReceivedDocumentTotalsResponse"));
var _GetPaymentAccountResponse = _interopRequireDefault(require("./model/GetPaymentAccountResponse"));
var _GetPaymentMethodResponse = _interopRequireDefault(require("./model/GetPaymentMethodResponse"));
var _GetProductResponse = _interopRequireDefault(require("./model/GetProductResponse"));
var _GetReceiptPreCreateInfoResponse = _interopRequireDefault(require("./model/GetReceiptPreCreateInfoResponse"));
var _GetReceiptResponse = _interopRequireDefault(require("./model/GetReceiptResponse"));
var _GetReceiptsMonthlyTotalsResponse = _interopRequireDefault(require("./model/GetReceiptsMonthlyTotalsResponse"));
var _GetReceivedDocumentPreCreateInfoResponse = _interopRequireDefault(require("./model/GetReceivedDocumentPreCreateInfoResponse"));
var _GetReceivedDocumentResponse = _interopRequireDefault(require("./model/GetReceivedDocumentResponse"));
var _GetSupplierResponse = _interopRequireDefault(require("./model/GetSupplierResponse"));
var _GetUserInfoResponse = _interopRequireDefault(require("./model/GetUserInfoResponse"));
var _GetUserInfoResponseEmailConfirmationState = _interopRequireDefault(require("./model/GetUserInfoResponseEmailConfirmationState"));
var _GetUserInfoResponseInfo = _interopRequireDefault(require("./model/GetUserInfoResponseInfo"));
var _GetVatTypeResponse = _interopRequireDefault(require("./model/GetVatTypeResponse"));
var _GetWebhooksSubscriptionResponse = _interopRequireDefault(require("./model/GetWebhooksSubscriptionResponse"));
var _IssuedDocument = _interopRequireDefault(require("./model/IssuedDocument"));
var _IssuedDocumentEiData = _interopRequireDefault(require("./model/IssuedDocumentEiData"));
var _IssuedDocumentExtraData = _interopRequireDefault(require("./model/IssuedDocumentExtraData"));
var _IssuedDocumentItemsListItem = _interopRequireDefault(require("./model/IssuedDocumentItemsListItem"));
var _IssuedDocumentOptions = _interopRequireDefault(require("./model/IssuedDocumentOptions"));
var _IssuedDocumentPaymentsListItem = _interopRequireDefault(require("./model/IssuedDocumentPaymentsListItem"));
var _IssuedDocumentPaymentsListItemPaymentTerms = _interopRequireDefault(require("./model/IssuedDocumentPaymentsListItemPaymentTerms"));
var _IssuedDocumentPreCreateInfo = _interopRequireDefault(require("./model/IssuedDocumentPreCreateInfo"));
var _IssuedDocumentPreCreateInfoDefaultValues = _interopRequireDefault(require("./model/IssuedDocumentPreCreateInfoDefaultValues"));
var _IssuedDocumentPreCreateInfoExtraDataDefaultValues = _interopRequireDefault(require("./model/IssuedDocumentPreCreateInfoExtraDataDefaultValues"));
var _IssuedDocumentPreCreateInfoItemsDefaultValues = _interopRequireDefault(require("./model/IssuedDocumentPreCreateInfoItemsDefaultValues"));
var _IssuedDocumentStatus = _interopRequireDefault(require("./model/IssuedDocumentStatus"));
var _IssuedDocumentTotals = _interopRequireDefault(require("./model/IssuedDocumentTotals"));
var _IssuedDocumentType = _interopRequireDefault(require("./model/IssuedDocumentType"));
var _JoinIssuedDocumentsResponse = _interopRequireDefault(require("./model/JoinIssuedDocumentsResponse"));
var _Language = _interopRequireDefault(require("./model/Language"));
var _ListArchiveCategoriesResponse = _interopRequireDefault(require("./model/ListArchiveCategoriesResponse"));
var _ListArchiveDocumentsResponse = _interopRequireDefault(require("./model/ListArchiveDocumentsResponse"));
var _ListArchiveDocumentsResponsePage = _interopRequireDefault(require("./model/ListArchiveDocumentsResponsePage"));
var _ListCashbookEntriesResponse = _interopRequireDefault(require("./model/ListCashbookEntriesResponse"));
var _ListCitiesResponse = _interopRequireDefault(require("./model/ListCitiesResponse"));
var _ListClientsResponse = _interopRequireDefault(require("./model/ListClientsResponse"));
var _ListClientsResponsePage = _interopRequireDefault(require("./model/ListClientsResponsePage"));
var _ListCostCentersResponse = _interopRequireDefault(require("./model/ListCostCentersResponse"));
var _ListCountriesResponse = _interopRequireDefault(require("./model/ListCountriesResponse"));
var _ListCurrenciesResponse = _interopRequireDefault(require("./model/ListCurrenciesResponse"));
var _ListDeliveryNotesDefaultCausalsResponse = _interopRequireDefault(require("./model/ListDeliveryNotesDefaultCausalsResponse"));
var _ListDetailedCountriesResponse = _interopRequireDefault(require("./model/ListDetailedCountriesResponse"));
var _ListEmailsResponse = _interopRequireDefault(require("./model/ListEmailsResponse"));
var _ListEmailsResponsePage = _interopRequireDefault(require("./model/ListEmailsResponsePage"));
var _ListF24Response = _interopRequireDefault(require("./model/ListF24Response"));
var _ListF24ResponseAggregatedData = _interopRequireDefault(require("./model/ListF24ResponseAggregatedData"));
var _ListF24ResponseAggregation = _interopRequireDefault(require("./model/ListF24ResponseAggregation"));
var _ListF24ResponsePage = _interopRequireDefault(require("./model/ListF24ResponsePage"));
var _ListIssuedDocumentsResponse = _interopRequireDefault(require("./model/ListIssuedDocumentsResponse"));
var _ListIssuedDocumentsResponsePage = _interopRequireDefault(require("./model/ListIssuedDocumentsResponsePage"));
var _ListLanguagesResponse = _interopRequireDefault(require("./model/ListLanguagesResponse"));
var _ListPaymentAccountsResponse = _interopRequireDefault(require("./model/ListPaymentAccountsResponse"));
var _ListPaymentMethodsResponse = _interopRequireDefault(require("./model/ListPaymentMethodsResponse"));
var _ListProductCategoriesResponse = _interopRequireDefault(require("./model/ListProductCategoriesResponse"));
var _ListProductsResponse = _interopRequireDefault(require("./model/ListProductsResponse"));
var _ListProductsResponsePage = _interopRequireDefault(require("./model/ListProductsResponsePage"));
var _ListReceiptsResponse = _interopRequireDefault(require("./model/ListReceiptsResponse"));
var _ListReceiptsResponsePage = _interopRequireDefault(require("./model/ListReceiptsResponsePage"));
var _ListReceivedDocumentCategoriesResponse = _interopRequireDefault(require("./model/ListReceivedDocumentCategoriesResponse"));
var _ListReceivedDocumentsResponse = _interopRequireDefault(require("./model/ListReceivedDocumentsResponse"));
var _ListReceivedDocumentsResponsePage = _interopRequireDefault(require("./model/ListReceivedDocumentsResponsePage"));
var _ListRevenueCentersResponse = _interopRequireDefault(require("./model/ListRevenueCentersResponse"));
var _ListSuppliersResponse = _interopRequireDefault(require("./model/ListSuppliersResponse"));
var _ListSuppliersResponsePage = _interopRequireDefault(require("./model/ListSuppliersResponsePage"));
var _ListTemplatesResponse = _interopRequireDefault(require("./model/ListTemplatesResponse"));
var _ListUnitsOfMeasureResponse = _interopRequireDefault(require("./model/ListUnitsOfMeasureResponse"));
var _ListUserCompaniesResponse = _interopRequireDefault(require("./model/ListUserCompaniesResponse"));
var _ListUserCompaniesResponseData = _interopRequireDefault(require("./model/ListUserCompaniesResponseData"));
var _ListVatTypesResponse = _interopRequireDefault(require("./model/ListVatTypesResponse"));
var _ListWebhooksSubscriptionsResponse = _interopRequireDefault(require("./model/ListWebhooksSubscriptionsResponse"));
var _ModifyArchiveDocumentRequest = _interopRequireDefault(require("./model/ModifyArchiveDocumentRequest"));
var _ModifyArchiveDocumentResponse = _interopRequireDefault(require("./model/ModifyArchiveDocumentResponse"));
var _ModifyCashbookEntryRequest = _interopRequireDefault(require("./model/ModifyCashbookEntryRequest"));
var _ModifyCashbookEntryResponse = _interopRequireDefault(require("./model/ModifyCashbookEntryResponse"));
var _ModifyClientRequest = _interopRequireDefault(require("./model/ModifyClientRequest"));
var _ModifyClientResponse = _interopRequireDefault(require("./model/ModifyClientResponse"));
var _ModifyF24Request = _interopRequireDefault(require("./model/ModifyF24Request"));
var _ModifyF24Response = _interopRequireDefault(require("./model/ModifyF24Response"));
var _ModifyIssuedDocumentRequest = _interopRequireDefault(require("./model/ModifyIssuedDocumentRequest"));
var _ModifyIssuedDocumentResponse = _interopRequireDefault(require("./model/ModifyIssuedDocumentResponse"));
var _ModifyPaymentAccountRequest = _interopRequireDefault(require("./model/ModifyPaymentAccountRequest"));
var _ModifyPaymentAccountResponse = _interopRequireDefault(require("./model/ModifyPaymentAccountResponse"));
var _ModifyPaymentMethodRequest = _interopRequireDefault(require("./model/ModifyPaymentMethodRequest"));
var _ModifyPaymentMethodResponse = _interopRequireDefault(require("./model/ModifyPaymentMethodResponse"));
var _ModifyProductRequest = _interopRequireDefault(require("./model/ModifyProductRequest"));
var _ModifyProductResponse = _interopRequireDefault(require("./model/ModifyProductResponse"));
var _ModifyReceiptRequest = _interopRequireDefault(require("./model/ModifyReceiptRequest"));
var _ModifyReceiptResponse = _interopRequireDefault(require("./model/ModifyReceiptResponse"));
var _ModifyReceivedDocumentRequest = _interopRequireDefault(require("./model/ModifyReceivedDocumentRequest"));
var _ModifyReceivedDocumentResponse = _interopRequireDefault(require("./model/ModifyReceivedDocumentResponse"));
var _ModifySupplierRequest = _interopRequireDefault(require("./model/ModifySupplierRequest"));
var _ModifySupplierResponse = _interopRequireDefault(require("./model/ModifySupplierResponse"));
var _ModifyVatTypeRequest = _interopRequireDefault(require("./model/ModifyVatTypeRequest"));
var _ModifyVatTypeResponse = _interopRequireDefault(require("./model/ModifyVatTypeResponse"));
var _ModifyWebhooksSubscriptionRequest = _interopRequireDefault(require("./model/ModifyWebhooksSubscriptionRequest"));
var _ModifyWebhooksSubscriptionResponse = _interopRequireDefault(require("./model/ModifyWebhooksSubscriptionResponse"));
var _MonthlyTotal = _interopRequireDefault(require("./model/MonthlyTotal"));
var _OriginalDocumentType = _interopRequireDefault(require("./model/OriginalDocumentType"));
var _Pagination = _interopRequireDefault(require("./model/Pagination"));
var _PaymentAccount = _interopRequireDefault(require("./model/PaymentAccount"));
var _PaymentAccountType = _interopRequireDefault(require("./model/PaymentAccountType"));
var _PaymentMethod = _interopRequireDefault(require("./model/PaymentMethod"));
var _PaymentMethodDetails = _interopRequireDefault(require("./model/PaymentMethodDetails"));
var _PaymentMethodType = _interopRequireDefault(require("./model/PaymentMethodType"));
var _PaymentTermsType = _interopRequireDefault(require("./model/PaymentTermsType"));
var _PermissionLevel = _interopRequireDefault(require("./model/PermissionLevel"));
var _Permissions = _interopRequireDefault(require("./model/Permissions"));
var _PermissionsFicIssuedDocumentsDetailed = _interopRequireDefault(require("./model/PermissionsFicIssuedDocumentsDetailed"));
var _Product = _interopRequireDefault(require("./model/Product"));
var _Receipt = _interopRequireDefault(require("./model/Receipt"));
var _ReceiptItemsListItem = _interopRequireDefault(require("./model/ReceiptItemsListItem"));
var _ReceiptPreCreateInfo = _interopRequireDefault(require("./model/ReceiptPreCreateInfo"));
var _ReceiptType = _interopRequireDefault(require("./model/ReceiptType"));
var _ReceivedDocument = _interopRequireDefault(require("./model/ReceivedDocument"));
var _ReceivedDocumentInfo = _interopRequireDefault(require("./model/ReceivedDocumentInfo"));
var _ReceivedDocumentInfoDefaultValues = _interopRequireDefault(require("./model/ReceivedDocumentInfoDefaultValues"));
var _ReceivedDocumentInfoItemsDefaultValues = _interopRequireDefault(require("./model/ReceivedDocumentInfoItemsDefaultValues"));
var _ReceivedDocumentItemsListItem = _interopRequireDefault(require("./model/ReceivedDocumentItemsListItem"));
var _ReceivedDocumentPaymentsListItem = _interopRequireDefault(require("./model/ReceivedDocumentPaymentsListItem"));
var _ReceivedDocumentPaymentsListItemPaymentTerms = _interopRequireDefault(require("./model/ReceivedDocumentPaymentsListItemPaymentTerms"));
var _ReceivedDocumentTotals = _interopRequireDefault(require("./model/ReceivedDocumentTotals"));
var _ReceivedDocumentType = _interopRequireDefault(require("./model/ReceivedDocumentType"));
var _ScheduleEmailRequest = _interopRequireDefault(require("./model/ScheduleEmailRequest"));
var _SendEInvoiceRequest = _interopRequireDefault(require("./model/SendEInvoiceRequest"));
var _SendEInvoiceRequestData = _interopRequireDefault(require("./model/SendEInvoiceRequestData"));
var _SendEInvoiceRequestOptions = _interopRequireDefault(require("./model/SendEInvoiceRequestOptions"));
var _SendEInvoiceResponse = _interopRequireDefault(require("./model/SendEInvoiceResponse"));
var _SendEInvoiceResponseData = _interopRequireDefault(require("./model/SendEInvoiceResponseData"));
var _SenderEmail = _interopRequireDefault(require("./model/SenderEmail"));
var _ShowTotalsMode = _interopRequireDefault(require("./model/ShowTotalsMode"));
var _Supplier = _interopRequireDefault(require("./model/Supplier"));
var _SupplierType = _interopRequireDefault(require("./model/SupplierType"));
var _TransformIssuedDocumentResponse = _interopRequireDefault(require("./model/TransformIssuedDocumentResponse"));
var _UploadArchiveAttachmentResponse = _interopRequireDefault(require("./model/UploadArchiveAttachmentResponse"));
var _UploadF24AttachmentResponse = _interopRequireDefault(require("./model/UploadF24AttachmentResponse"));
var _UploadIssuedDocumentAttachmentResponse = _interopRequireDefault(require("./model/UploadIssuedDocumentAttachmentResponse"));
var _UploadReceivedDocumentAttachmentResponse = _interopRequireDefault(require("./model/UploadReceivedDocumentAttachmentResponse"));
var _User = _interopRequireDefault(require("./model/User"));
var _UserCompanyRole = _interopRequireDefault(require("./model/UserCompanyRole"));
var _VatItem = _interopRequireDefault(require("./model/VatItem"));
var _VatKind = _interopRequireDefault(require("./model/VatKind"));
var _VatType = _interopRequireDefault(require("./model/VatType"));
var _VerifyEInvoiceXmlErrorResponse = _interopRequireDefault(require("./model/VerifyEInvoiceXmlErrorResponse"));
var _VerifyEInvoiceXmlErrorResponseError = _interopRequireDefault(require("./model/VerifyEInvoiceXmlErrorResponseError"));
var _VerifyEInvoiceXmlErrorResponseErrorValidationResult = _interopRequireDefault(require("./model/VerifyEInvoiceXmlErrorResponseErrorValidationResult"));
var _VerifyEInvoiceXmlErrorResponseExtra = _interopRequireDefault(require("./model/VerifyEInvoiceXmlErrorResponseExtra"));
var _VerifyEInvoiceXmlResponse = _interopRequireDefault(require("./model/VerifyEInvoiceXmlResponse"));
var _VerifyEInvoiceXmlResponseData = _interopRequireDefault(require("./model/VerifyEInvoiceXmlResponseData"));
var _WebhooksSubscription = _interopRequireDefault(require("./model/WebhooksSubscription"));
var _WebhooksSubscriptionConfig = _interopRequireDefault(require("./model/WebhooksSubscriptionConfig"));
var _WebhooksSubscriptionMapping = _interopRequireDefault(require("./model/WebhooksSubscriptionMapping"));
var _ArchiveApi = _interopRequireDefault(require("./api/ArchiveApi"));
var _CashbookApi = _interopRequireDefault(require("./api/CashbookApi"));
var _ClientsApi = _interopRequireDefault(require("./api/ClientsApi"));
var _CompaniesApi = _interopRequireDefault(require("./api/CompaniesApi"));
var _EmailsApi = _interopRequireDefault(require("./api/EmailsApi"));
var _InfoApi = _interopRequireDefault(require("./api/InfoApi"));
var _IssuedDocumentsApi = _interopRequireDefault(require("./api/IssuedDocumentsApi"));
var _IssuedEInvoicesApi = _interopRequireDefault(require("./api/IssuedEInvoicesApi"));
var _ProductsApi = _interopRequireDefault(require("./api/ProductsApi"));
var _ReceiptsApi = _interopRequireDefault(require("./api/ReceiptsApi"));
var _ReceivedDocumentsApi = _interopRequireDefault(require("./api/ReceivedDocumentsApi"));
var _SettingsApi = _interopRequireDefault(require("./api/SettingsApi"));
var _SuppliersApi = _interopRequireDefault(require("./api/SuppliersApi"));
var _TaxesApi = _interopRequireDefault(require("./api/TaxesApi"));
var _UserApi = _interopRequireDefault(require("./api/UserApi"));
var _WebhooksApi = _interopRequireDefault(require("./api/WebhooksApi"));
var _Condition = _interopRequireDefault(require("./filter/Condition"));
var _Filter = _interopRequireDefault(require("./filter/Filter"));
var _Expression = _interopRequireDefault(require("./filter/Expression"));
var _Conjunction = _interopRequireDefault(require("./filter/Conjunction"));
var _Disjunction = _interopRequireDefault(require("./filter/Disjunction"));
var _Operator = _interopRequireDefault(require("./filter/Operator"));
var _OAuth = _interopRequireDefault(require("./oauth2/OAuth2"));
var _OAuth2AuthorizationCodeTokenResponse = _interopRequireDefault(require("./oauth2/OAuth2AuthorizationCodeTokenResponse"));
var _OAuth2AuthorizationCodeParams = _interopRequireDefault(require("./oauth2/OAuth2AuthorizationCodeParams"));
var _Scope = _interopRequireDefault(require("./oauth2/Scope"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
